<!DOCTYPE html>
<head>
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Genome Graphics  | Manual</title>
<link rel="stylesheet" href="/CSHL_RNAseq/css/syntax.css">


<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="css/bootstrap.min.css">-->
<link rel="stylesheet" href="/CSHL_RNAseq/css/modern-business.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/lavish-bootstrap.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/customstyles.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/theme-blue.css">

<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="/CSHL_RNAseq/js/jquery.navgoco.min.js"></script>


<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="/CSHL_RNAseq/js/toc.js"></script>
<script src="/CSHL_RNAseq/js/customscripts.js"></script>

<link rel="shortcut icon" href="/CSHL_RNAseq/images/favicon.ico">

<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->

<link rel="alternate" type="application/rss+xml" title="" href="http://idratherbewriting.com/CSHL_RNAseq/feed.xml">




 
    <script>
        $(document).ready(function() {
            // Initialize navgoco with default options
            $("#mysidebar").navgoco({
                caretHtml: '',
                accordion: true,
                openClass: 'active', // open
                save: false, // leave false or nav highlighting doesn't work right
                cookie: {
                    name: 'navgoco',
                    expires: false,
                    path: '/'
                },
                slide: {
                    duration: 400,
                    easing: 'swing'
                }
            });

            $("#collapseAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', false);
            });

            $("#expandAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', true);
            });

        });

    </script>
    <script>
        $(function () {
            $('[data-toggle="tooltip"]').tooltip()
        })
    </script>
    

</head>
<body>
<!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="fa fa-home fa-lg navbar-brand" href="/CSHL_RNAseq/">&nbsp;<span class="projectTitle"> 2016 CSHL Course - Plant Sciences</span></a>
        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                
                
                
                <li><a href="https://meetings.cshl.edu/courses.aspx?course=c-plan&year=16" target="_blank">CSHL</a></li>
                
                
                
                <li><a href="https://github.com/tgirke/CSHL_RNAseq/tree/gh-pages" target="_blank">Github Repo</a></li>
                
                
                
                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                
                
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">R on CRAN</a></li>
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        <li><a href="https://support.rstudio.com/hc/en-us/articles/200711853-Keyboard-Shortcuts" target="_blank">RStudio Shortcuts</a></li>
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                    </ul>
                </li>
                
                
                
			<!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "Manual feedback";
var body = "I have some feedback about the Genome Graphics page: ";
body += window.location.href;
body += "";
var uri = "mailto:thomas.girke@ucr.edu?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>

		
                <!--comment out this block if you want to hide search-->
                <li>
                    <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="/CSHL_RNAseq/js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                            SimpleJekyllSearch.init({
                                searchInput: document.getElementById('search-input'),
                                resultsContainer: document.getElementById('results-container'),
                                dataSource: '/CSHL_RNAseq/search.json',
                                searchResultTemplate: '<li><a href="{url}" title="Genome Graphics">{title}</a></li>',
                    noResultsText: 'No results found.',
                            limit: 10,
                            fuzzy: true,
                    })
                    </script>
                    <!--end search-->
                </li>
            </ul>
        </div>
        </div>
        <!-- /.container -->
</nav>
<!-- Page Content -->
<div class="container">
    <div class="col-lg-12">&nbsp;</div>
    <!-- Content Row -->
    <div class="row">
        <!-- Sidebar Column -->
        <div class="col-md-3">

          












<ul id="mysidebar" class="nav">
    <li class="sidebarTitle">RNA-Seq </li>
    
    
    
        
    
    <li>
        <a href="#">Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/">Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_requirements/">Requirements</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Slides</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_slides/">Slides</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Introduction to R</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_01/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_02/">2. R Package Repositories</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_03/">3. Installation of R and Add-on Packages</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_04/">4. Getting Around</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_05/">5. Basic Syntax</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_06/">6. Data Types</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_07/">7. Data objects</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_08/">8. Important Utilities</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_09/">9. Operators and Calculations</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_10/">10. Reading and Writing External Data</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_11/">11. Useful R Functions</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_12/">12. SQLite Databases</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_13/">13. R Markdown</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_14/">14. Session Info</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_15/">15. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">R Graphics</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_1/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_2/">2. Base Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_3/">3. Grid Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_4/">4. lattice Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_5/">5. ggplot2 Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_6/">6. Specialty Graphics</a></li>
            
            
            
            
            
            
            <li class="active"><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_7/">7. Genome Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_8/">8. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">NGS Workflow Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_1/">1. Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_2/">2. Getting Started</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_3/">3. More detail</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_4/">4. Workflow demo</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_5/">5. Version information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_6/">6. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">RNA-Seq Workflow</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_01/">1. Introduction</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_02/">2. Sample definitions and environment settings</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_03/">3. Read preprocessing</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_04/">4. Alignments</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_05/">5. Read distribution across genomic features</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_06/">6. Read quantification per annotation range</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_07/">7. Analysis of differentially expressed genes with edgeR</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_08/">8. GO term enrichment analysis of DEGs</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_09/">9. Clustering and heat maps</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_10/">10. Render report in HTML and PDF format</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_11/">11. Version Information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_12/">12. Funding</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_13/">13. References</a></li>
            
            
            
            
        </ul>
        
        
        
        <!-- if you aren't using the accordion, uncomment this block:
           <p class="external">
               <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
           </p>
           -->
    </li>
</ul>
</div>

<!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
<script>$("li.active").parents('li').toggleClass("active");</script>
    <!-- Content Column -->
    <div class="col-md-9">
        <div class="post-header">
   <h1 class="post-title-main">Genome Graphics</h1>
</div>



<div class="post-content">

   

    
    
<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>

<div id="toc"></div>

    

    

    

    

  <h2 id="ggbio"><code class="highlighter-rouge">ggbio</code></h2>

<ul>
  <li>What is <code class="highlighter-rouge">ggbio</code>?
    <ul>
      <li>A programmable genome browser environment</li>
    </ul>
  </li>
  <li>Genome broswer concepts
    <ul>
      <li>A genome browser is a visulalization tool for plotting different types of genomic data in separate tracks along chromosomes.</li>
      <li>The <code class="highlighter-rouge">ggbio</code> package (Yin et al., 2012) facilitates plotting of complex genome data objects, such as read alignments (SAM/BAM), genomic context/annotation information (gff/txdb), variant calls (VCF/BCF), and more. To easily compare these data sets, it extends the faceting facility of <code class="highlighter-rouge">ggplot2</code> to genome browser-like tracks.</li>
      <li>Most of the core object types for handling genomic data with R/Bioconductor are supported: <code class="highlighter-rouge">GRanges</code>, <code class="highlighter-rouge">GAlignments</code>, <code class="highlighter-rouge">VCF</code>, etc. For more details, see Table 1.1 of the <code class="highlighter-rouge">ggbio</code> vignette <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ggbio/inst/doc/ggbio.pdf">here</a>.</li>
      <li><code class="highlighter-rouge">ggbio</code>â€™s convenience plotting function is <code class="highlighter-rouge">autoplot</code>. For more customizable plots, one can use the generic <code class="highlighter-rouge">ggplot</code> function.</li>
      <li>Apart from the standard <code class="highlighter-rouge">ggplot2</code> plotting components, <code class="highlighter-rouge">ggbio</code> defines serval new components useful for genomic data visualization. A detailed list is given in Table 1.2 of the vignette <a href="http://www.bioconductor.org/packages/release/bioc/vignettes/ggbio/inst/doc/ggbio.pdf">here</a>.</li>
      <li>Useful web sites:
        <ul>
          <li><a href="http://www.tengfei.name/ggbio/docs/">ggbio manual</a></li>
          <li><a href="http://www.tengfei.name/ggbio/docs/man/">ggbio functions</a></li>
          <li><a href="http://www.tengfei.name/ggbio/docs/man/autoplot-method.html">autoplot demo</a></li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<h3 id="tracks-aligning-plots-along-chromosomes">Tracks: aligning plots along chromosomes</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">ggbio</span><span class="p">)</span>
<span class="n">df1</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">time</span> <span class="o">=</span> <span class="m">1</span><span class="o">:</span><span class="m">100</span><span class="p">,</span> <span class="n">score</span> <span class="o">=</span> <span class="n">sin</span><span class="p">((</span><span class="m">1</span><span class="o">:</span><span class="m">100</span><span class="p">)</span><span class="o">/</span><span class="m">20</span><span class="p">)</span><span class="o">*</span><span class="m">10</span><span class="p">)</span>
<span class="n">p1</span> <span class="o">&lt;-</span> <span class="n">qplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">df1</span><span class="p">,</span> <span class="n">x</span> <span class="o">=</span> <span class="n">time</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">score</span><span class="p">,</span> <span class="n">geom</span> <span class="o">=</span> <span class="s2">"line"</span><span class="p">)</span>
<span class="n">df2</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">time</span> <span class="o">=</span> <span class="m">30</span><span class="o">:</span><span class="m">120</span><span class="p">,</span> <span class="n">score</span> <span class="o">=</span> <span class="n">sin</span><span class="p">((</span><span class="m">30</span><span class="o">:</span><span class="m">120</span><span class="p">)</span><span class="o">/</span><span class="m">20</span><span class="p">)</span><span class="o">*</span><span class="m">10</span><span class="p">,</span> <span class="n">value</span> <span class="o">=</span> <span class="n">rnorm</span><span class="p">(</span><span class="m">120-30</span> <span class="m">+1</span><span class="p">))</span>
<span class="n">p2</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">data</span> <span class="o">=</span> <span class="n">df2</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">x</span> <span class="o">=</span> <span class="n">time</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">score</span><span class="p">))</span> <span class="o">+</span> <span class="n">geom_line</span><span class="p">()</span> <span class="o">+</span> <span class="n">geom_point</span><span class="p">(</span><span class="n">size</span> <span class="o">=</span> <span class="m">2</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="n">value</span><span class="p">))</span>
<span class="n">tracks</span><span class="p">(</span><span class="n">time1</span> <span class="o">=</span> <span class="n">p1</span><span class="p">,</span> <span class="n">time2</span> <span class="o">=</span> <span class="n">p2</span><span class="p">)</span> <span class="o">+</span> <span class="n">xlim</span><span class="p">(</span><span class="m">1</span><span class="p">,</span> <span class="m">40</span><span class="p">)</span> <span class="o">+</span> <span class="n">theme_tracks_sunset</span><span class="p">()</span></code></pre></figure>

<p><img src="../Rgraphics_files/ggbio_align-1.png" alt="" /></p>

<h3 id="plotting-genomic-ranges">Plotting genomic ranges</h3>

<p><code class="highlighter-rouge">GRanges</code> objects are essential for storing alignment or annotation ranges in R/Bioconductor. The following creates a sample <code class="highlighter-rouge">GRanges</code> object and plots its content.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">GenomicRanges</span><span class="p">)</span>
<span class="n">set.seed</span><span class="p">(</span><span class="m">1</span><span class="p">);</span> <span class="n">N</span> <span class="o">&lt;-</span> <span class="m">100</span><span class="p">;</span> <span class="n">gr</span> <span class="o">&lt;-</span> <span class="n">GRanges</span><span class="p">(</span><span class="n">seqnames</span> <span class="o">=</span> <span class="n">sample</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="s2">"chr1"</span><span class="p">,</span> <span class="s2">"chr2"</span><span class="p">,</span> <span class="s2">"chr3"</span><span class="p">),</span> <span class="n">size</span> <span class="o">=</span> <span class="n">N</span><span class="p">,</span> <span class="n">replace</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">IRanges</span><span class="p">(</span><span class="n">start</span> <span class="o">=</span> <span class="n">sample</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">300</span><span class="p">,</span> <span class="n">size</span> <span class="o">=</span> <span class="n">N</span><span class="p">,</span> <span class="n">replace</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">width</span> <span class="o">=</span> <span class="n">sample</span><span class="p">(</span><span class="m">70</span><span class="o">:</span><span class="m">75</span><span class="p">,</span> <span class="n">size</span> <span class="o">=</span> <span class="n">N</span><span class="p">,</span><span class="n">replace</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">)),</span> <span class="n">strand</span> <span class="o">=</span> <span class="n">sample</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="s2">"+"</span><span class="p">,</span> <span class="s2">"-"</span><span class="p">),</span> <span class="n">size</span> <span class="o">=</span> <span class="n">N</span><span class="p">,</span> <span class="n">replace</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">value</span> <span class="o">=</span> <span class="n">rnorm</span><span class="p">(</span><span class="n">N</span><span class="p">,</span> <span class="m">10</span><span class="p">,</span> <span class="m">3</span><span class="p">),</span> <span class="n">score</span> <span class="o">=</span> <span class="n">rnorm</span><span class="p">(</span><span class="n">N</span><span class="p">,</span> <span class="m">100</span><span class="p">,</span> <span class="m">30</span><span class="p">),</span> <span class="n">sample</span> <span class="o">=</span> <span class="n">sample</span><span class="p">(</span><span class="n">c</span><span class="p">(</span><span class="s2">"Normal"</span><span class="p">,</span> <span class="s2">"Tumor"</span><span class="p">),</span> <span class="n">size</span> <span class="o">=</span> <span class="n">N</span><span class="p">,</span> <span class="n">replace</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">),</span> <span class="n">pair</span> <span class="o">=</span> <span class="n">sample</span><span class="p">(</span><span class="n">letters</span><span class="p">,</span> <span class="n">size</span> <span class="o">=</span> <span class="n">N</span><span class="p">,</span> <span class="n">replace</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">))</span>
<span class="n">autoplot</span><span class="p">(</span><span class="n">gr</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="n">strand</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="n">strand</span><span class="p">),</span> <span class="n">facets</span> <span class="o">=</span> <span class="n">strand</span> <span class="o">~</span> <span class="n">seqnames</span><span class="p">)</span></code></pre></figure>

<p><img src="../Rgraphics_files/ggbio_granges-1.png" alt="" /></p>

<h3 id="plotting-coverage">Plotting coverage</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">autoplot</span><span class="p">(</span><span class="n">gr</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">color</span> <span class="o">=</span> <span class="n">strand</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="n">strand</span><span class="p">),</span> <span class="n">facets</span> <span class="o">=</span> <span class="n">strand</span> <span class="o">~</span> <span class="n">seqnames</span><span class="p">,</span> <span class="n">stat</span> <span class="o">=</span> <span class="s2">"coverage"</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## Scale for 'x' is already present. Adding another scale for 'x', which will replace the existing
## scale.</code></pre></figure>

<p><img src="../Rgraphics_files/ggbio_coverage-1.png" alt="" /></p>

<h3 id="mirrored-coverage">Mirrored coverage</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">pos</span> <span class="o">&lt;-</span> <span class="n">sapply</span><span class="p">(</span><span class="n">coverage</span><span class="p">(</span><span class="n">gr</span><span class="p">[</span><span class="n">strand</span><span class="p">(</span><span class="n">gr</span><span class="p">)</span><span class="o">==</span><span class="s2">"+"</span><span class="p">]),</span> <span class="n">as.numeric</span><span class="p">)</span>
<span class="n">pos</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">Chr</span><span class="o">=</span><span class="n">rep</span><span class="p">(</span><span class="n">names</span><span class="p">(</span><span class="n">pos</span><span class="p">),</span> <span class="n">sapply</span><span class="p">(</span><span class="n">pos</span><span class="p">,</span> <span class="n">length</span><span class="p">)),</span> <span class="n">Strand</span><span class="o">=</span><span class="n">rep</span><span class="p">(</span><span class="s2">"+"</span><span class="p">,</span> <span class="n">length</span><span class="p">(</span><span class="n">unlist</span><span class="p">(</span><span class="n">pos</span><span class="p">))),</span> <span class="n">Position</span><span class="o">=</span><span class="n">unlist</span><span class="p">(</span><span class="n">sapply</span><span class="p">(</span><span class="n">pos</span><span class="p">,</span> <span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="m">1</span><span class="o">:</span><span class="n">length</span><span class="p">(</span><span class="n">x</span><span class="p">))),</span> <span class="n">Coverage</span><span class="o">=</span><span class="n">as.numeric</span><span class="p">(</span><span class="n">unlist</span><span class="p">(</span><span class="n">pos</span><span class="p">)))</span>
<span class="n">neg</span> <span class="o">&lt;-</span> <span class="n">sapply</span><span class="p">(</span><span class="n">coverage</span><span class="p">(</span><span class="n">gr</span><span class="p">[</span><span class="n">strand</span><span class="p">(</span><span class="n">gr</span><span class="p">)</span><span class="o">==</span><span class="s2">"-"</span><span class="p">]),</span> <span class="n">as.numeric</span><span class="p">)</span>
<span class="n">neg</span> <span class="o">&lt;-</span> <span class="n">data.frame</span><span class="p">(</span><span class="n">Chr</span><span class="o">=</span><span class="n">rep</span><span class="p">(</span><span class="n">names</span><span class="p">(</span><span class="n">neg</span><span class="p">),</span> <span class="n">sapply</span><span class="p">(</span><span class="n">neg</span><span class="p">,</span> <span class="n">length</span><span class="p">)),</span> <span class="n">Strand</span><span class="o">=</span><span class="n">rep</span><span class="p">(</span><span class="s2">"-"</span><span class="p">,</span> <span class="n">length</span><span class="p">(</span><span class="n">unlist</span><span class="p">(</span><span class="n">neg</span><span class="p">))),</span> <span class="n">Position</span><span class="o">=</span><span class="n">unlist</span><span class="p">(</span><span class="n">sapply</span><span class="p">(</span><span class="n">neg</span><span class="p">,</span> <span class="k">function</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="m">1</span><span class="o">:</span><span class="n">length</span><span class="p">(</span><span class="n">x</span><span class="p">))),</span> <span class="n">Coverage</span><span class="o">=-</span><span class="n">as.numeric</span><span class="p">(</span><span class="n">unlist</span><span class="p">(</span><span class="n">neg</span><span class="p">)))</span>
<span class="n">covdf</span> <span class="o">&lt;-</span> <span class="n">rbind</span><span class="p">(</span><span class="n">pos</span><span class="p">,</span> <span class="n">neg</span><span class="p">)</span>
<span class="n">p</span> <span class="o">&lt;-</span> <span class="n">ggplot</span><span class="p">(</span><span class="n">covdf</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">Position</span><span class="p">,</span> <span class="n">Coverage</span><span class="p">,</span> <span class="n">fill</span><span class="o">=</span><span class="n">Strand</span><span class="p">))</span> <span class="o">+</span> 
	    <span class="n">geom_bar</span><span class="p">(</span><span class="n">stat</span><span class="o">=</span><span class="s2">"identity"</span><span class="p">,</span> <span class="n">position</span><span class="o">=</span><span class="s2">"identity"</span><span class="p">)</span> <span class="o">+</span> <span class="n">facet_wrap</span><span class="p">(</span><span class="o">~</span><span class="n">Chr</span><span class="p">)</span>
<span class="n">p</span></code></pre></figure>

<p><img src="../Rgraphics_files/ggbio_mirrored_coverage-1.png" alt="" /></p>

<h2 id="circular-genome-plots">Circular genome plots</h2>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">ggplot</span><span class="p">(</span><span class="n">gr</span><span class="p">)</span> <span class="o">+</span> <span class="n">layout_circle</span><span class="p">(</span><span class="n">aes</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="n">seqnames</span><span class="p">),</span> <span class="n">geom</span> <span class="o">=</span> <span class="s2">"rect"</span><span class="p">)</span></code></pre></figure>

<p><img src="../Rgraphics_files/ggbio_circular1-1.png" alt="" /></p>

<p>More complex circular example</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">seqlengths</span><span class="p">(</span><span class="n">gr</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="m">400</span><span class="p">,</span> <span class="m">500</span><span class="p">,</span> <span class="m">700</span><span class="p">)</span>
<span class="n">values</span><span class="p">(</span><span class="n">gr</span><span class="p">)</span><span class="o">$</span><span class="n">to.gr</span> <span class="o">&lt;-</span> <span class="n">gr</span><span class="p">[</span><span class="n">sample</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="n">length</span><span class="p">(</span><span class="n">gr</span><span class="p">),</span> <span class="n">size</span> <span class="o">=</span> <span class="n">length</span><span class="p">(</span><span class="n">gr</span><span class="p">))]</span>
<span class="n">idx</span> <span class="o">&lt;-</span> <span class="n">sample</span><span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="n">length</span><span class="p">(</span><span class="n">gr</span><span class="p">),</span> <span class="n">size</span> <span class="o">=</span> <span class="m">50</span><span class="p">)</span>
<span class="n">gr</span> <span class="o">&lt;-</span> <span class="n">gr</span><span class="p">[</span><span class="n">idx</span><span class="p">]</span>
<span class="n">ggplot</span><span class="p">()</span> <span class="o">+</span> <span class="n">layout_circle</span><span class="p">(</span><span class="n">gr</span><span class="p">,</span> <span class="n">geom</span> <span class="o">=</span> <span class="s2">"ideo"</span><span class="p">,</span> <span class="n">fill</span> <span class="o">=</span> <span class="s2">"gray70"</span><span class="p">,</span> <span class="n">radius</span> <span class="o">=</span> <span class="m">7</span><span class="p">,</span> <span class="n">trackWidth</span> <span class="o">=</span> <span class="m">3</span><span class="p">)</span> <span class="o">+</span>
  <span class="n">layout_circle</span><span class="p">(</span><span class="n">gr</span><span class="p">,</span> <span class="n">geom</span> <span class="o">=</span> <span class="s2">"bar"</span><span class="p">,</span> <span class="n">radius</span> <span class="o">=</span> <span class="m">10</span><span class="p">,</span> <span class="n">trackWidth</span> <span class="o">=</span> <span class="m">4</span><span class="p">,</span>
                <span class="n">aes</span><span class="p">(</span><span class="n">fill</span> <span class="o">=</span> <span class="n">score</span><span class="p">,</span> <span class="n">y</span> <span class="o">=</span> <span class="n">score</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">layout_circle</span><span class="p">(</span><span class="n">gr</span><span class="p">,</span> <span class="n">geom</span> <span class="o">=</span> <span class="s2">"point"</span><span class="p">,</span> <span class="n">color</span> <span class="o">=</span> <span class="s2">"red"</span><span class="p">,</span> <span class="n">radius</span> <span class="o">=</span> <span class="m">14</span><span class="p">,</span>
                <span class="n">trackWidth</span> <span class="o">=</span> <span class="m">3</span><span class="p">,</span> <span class="n">grid</span> <span class="o">=</span> <span class="n">TRUE</span><span class="p">,</span> <span class="n">aes</span><span class="p">(</span><span class="n">y</span> <span class="o">=</span> <span class="n">score</span><span class="p">))</span> <span class="o">+</span>
  <span class="n">layout_circle</span><span class="p">(</span><span class="n">gr</span><span class="p">,</span> <span class="n">geom</span> <span class="o">=</span> <span class="s2">"link"</span><span class="p">,</span> <span class="n">linked.to</span> <span class="o">=</span> <span class="s2">"to.gr"</span><span class="p">,</span> <span class="n">radius</span> <span class="o">=</span> <span class="m">6</span><span class="p">,</span> <span class="n">trackWidth</span> <span class="o">=</span> <span class="m">1</span><span class="p">)</span></code></pre></figure>

<center><img title="vim-r" src="../Rgraphics_files/circularplot.png" /></center>

<h3 id="alignments-and-variants">Alignments and variants</h3>

<p>To make the following example work, please download and unpack this <a href="http://biocluster.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_12_16_2013/Rgraphics/data.zip">data archive</a> containing GFF, BAM and VCF sample files.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">rtracklayer</span><span class="p">);</span> <span class="n">library</span><span class="p">(</span><span class="n">GenomicFeatures</span><span class="p">);</span> <span class="n">library</span><span class="p">(</span><span class="n">Rsamtools</span><span class="p">);</span> <span class="n">library</span><span class="p">(</span><span class="n">GenomicAlignments</span><span class="p">);</span> <span class="n">library</span><span class="p">(</span><span class="n">VariantAnnotation</span><span class="p">)</span>
<span class="n">ga</span> <span class="o">&lt;-</span> <span class="n">readGAlignments</span><span class="p">(</span><span class="s2">"./data/SRR064167.fastq.bam"</span><span class="p">,</span> <span class="n">use.names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">param</span><span class="o">=</span><span class="n">ScanBamParam</span><span class="p">(</span><span class="n">which</span><span class="o">=</span><span class="n">GRanges</span><span class="p">(</span><span class="s2">"Chr5"</span><span class="p">,</span> <span class="n">IRanges</span><span class="p">(</span><span class="m">4000</span><span class="p">,</span> <span class="m">8000</span><span class="p">))))</span>
<span class="n">p1</span> <span class="o">&lt;-</span> <span class="n">autoplot</span><span class="p">(</span><span class="n">ga</span><span class="p">,</span> <span class="n">geom</span> <span class="o">=</span> <span class="s2">"rect"</span><span class="p">)</span>
<span class="n">p2</span> <span class="o">&lt;-</span> <span class="n">autoplot</span><span class="p">(</span><span class="n">ga</span><span class="p">,</span> <span class="n">geom</span> <span class="o">=</span> <span class="s2">"line"</span><span class="p">,</span> <span class="n">stat</span> <span class="o">=</span> <span class="s2">"coverage"</span><span class="p">)</span>
<span class="n">vcf</span> <span class="o">&lt;-</span> <span class="n">readVcf</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s2">"data/varianttools_gnsap.vcf"</span><span class="p">,</span> <span class="n">genome</span><span class="o">=</span><span class="s2">"ATH1"</span><span class="p">)</span>
<span class="n">p3</span> <span class="o">&lt;-</span> <span class="n">autoplot</span><span class="p">(</span><span class="n">vcf</span><span class="p">[</span><span class="n">seqnames</span><span class="p">(</span><span class="n">vcf</span><span class="p">)</span><span class="o">==</span><span class="s2">"Chr5"</span><span class="p">],</span> <span class="n">type</span> <span class="o">=</span> <span class="s2">"fixed"</span><span class="p">)</span> <span class="o">+</span> <span class="n">xlim</span><span class="p">(</span><span class="m">4000</span><span class="p">,</span> <span class="m">8000</span><span class="p">)</span> <span class="o">+</span> <span class="n">theme</span><span class="p">(</span><span class="n">legend.position</span> <span class="o">=</span> <span class="s2">"none"</span><span class="p">,</span> <span class="n">axis.text.y</span> <span class="o">=</span> <span class="n">element_blank</span><span class="p">(),</span> <span class="n">axis.ticks.y</span><span class="o">=</span><span class="n">element_blank</span><span class="p">())</span>
<span class="n">txdb</span> <span class="o">&lt;-</span> <span class="n">makeTxDbFromGFF</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="s2">"./data/TAIR10_GFF3_trunc.gff"</span><span class="p">,</span> <span class="n">format</span><span class="o">=</span><span class="s2">"gff3"</span><span class="p">)</span>
<span class="n">p4</span> <span class="o">&lt;-</span> <span class="n">autoplot</span><span class="p">(</span><span class="n">txdb</span><span class="p">,</span> <span class="n">which</span><span class="o">=</span><span class="n">GRanges</span><span class="p">(</span><span class="s2">"Chr5"</span><span class="p">,</span> <span class="n">IRanges</span><span class="p">(</span><span class="m">4000</span><span class="p">,</span> <span class="m">8000</span><span class="p">)),</span> <span class="n">names.expr</span> <span class="o">=</span> <span class="s2">"gene_id"</span><span class="p">)</span>
<span class="n">tracks</span><span class="p">(</span><span class="n">Reads</span><span class="o">=</span><span class="n">p1</span><span class="p">,</span> <span class="n">Coverage</span><span class="o">=</span><span class="n">p2</span><span class="p">,</span> <span class="n">Variant</span><span class="o">=</span><span class="n">p3</span><span class="p">,</span> <span class="n">Transcripts</span><span class="o">=</span><span class="n">p4</span><span class="p">,</span> <span class="n">heights</span> <span class="o">=</span> <span class="n">c</span><span class="p">(</span><span class="m">0.3</span><span class="p">,</span> <span class="m">0.2</span><span class="p">,</span> <span class="m">0.1</span><span class="p">,</span> <span class="m">0.35</span><span class="p">))</span> <span class="o">+</span> <span class="n">ylab</span><span class="p">(</span><span class="s2">""</span><span class="p">)</span></code></pre></figure>

<p><img src="../Rgraphics_files/ggbio_align_variants-1.png" alt="" /></p>

<h2 id="additional-examples">Additional examples</h2>

<p>See <code class="highlighter-rouge">autoplot</code> demo <a href="http://www.tengfei.name/ggbio/docs/man/autoplot-method.html">here</a></p>

<h2 id="additional-genome-graphics">Additional genome graphics</h2>

<ul>
  <li><a href="http://www.bioconductor.org/packages/devel/bioc/html/Gviz.html"><code class="highlighter-rouge">Gviz</code></a></li>
  <li><a href="http://cran.us.r-project.org/web/packages/RCircos/index.html"><code class="highlighter-rouge">RCircos</code></a> (Zhang et al., 2013)</li>
  <li><a href="http://bioconductor.org/packages/release/bioc/html/GenomeGraphs.html"><code class="highlighter-rouge">Genome Graphs</code></a></li>
  <li><a href="http://genoplotr.r-forge.r-project.org/"><code class="highlighter-rouge">genoPlotR</code></a></li>
</ul>

<h2 id="genome-browser-igv">Genome Browser: IGV</h2>

<p>View genome data in IGV</p>

<ul>
  <li>Download and open <a href="http://www.broadinstitute.org/igv/download">IGV</a></li>
  <li>Select in menu in top left corner <em>A. thaliana</em> (TAIR10)</li>
  <li>Upload the following indexed/sorted Bam files with <code class="highlighter-rouge">File -&gt; Load from URL...</code></li>
</ul>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_6_10_2012/Rrnaseq/results/SRR064154.fastq.bam
http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_6_10_2012/Rrnaseq/results/SRR064155.fastq.bam
http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_6_10_2012/Rrnaseq/results/SRR064166.fastq.bam
http://faculty.ucr.edu/~tgirke/HTML_Presentations/Manuals/Workshop_Dec_6_10_2012/Rrnaseq/results/SRR064167.fastq.bam</code></pre></figure>

<ul>
  <li>To view area of interest, enter its coordinates <code class="highlighter-rouge">Chr1:49,457-51,457</code> in position menu on top.</li>
</ul>

<center><img title="vim-r" src="../Rgraphics_files/igv_peak_3.png" /></center>

<h3 id="create-symbolic-links">Create symbolic links</h3>

<p>For viewing BAM files in IGV as part of <code class="highlighter-rouge">systemPipeR</code> workflows.</p>

<ul>
  <li><code class="highlighter-rouge">systemPipeR</code>: utilities for building NGS analysis <a href="https://github.com/tgirke/systemPipeR">pipelines</a></li>
</ul>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="s2">"systemPipeR"</span><span class="p">)</span>
<span class="n">symLink2bam</span><span class="p">(</span><span class="n">sysargs</span><span class="o">=</span><span class="n">args</span><span class="p">,</span> <span class="n">htmldir</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="s2">"~/.html/"</span><span class="p">,</span> <span class="s2">"somedir/"</span><span class="p">),</span> 
            <span class="n">urlbase</span><span class="o">=</span><span class="s2">"http://myserver.edu/~username/"</span><span class="p">,</span>
            <span class="n">urlfile</span><span class="o">=</span><span class="s2">"IGVurl.txt"</span><span class="p">)</span></code></pre></figure>

<h3 id="controlling-igv-from-r">Controlling IGV from R</h3>

<p>Note this may not work on all systems.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">SRAdb</span><span class="p">)</span>
<span class="n">startIGV</span><span class="p">(</span><span class="s2">"lm"</span><span class="p">)</span>
<span class="n">sock</span> <span class="o">&lt;-</span> <span class="n">IGVsocket</span><span class="p">()</span>
<span class="n">session</span> <span class="o">&lt;-</span> <span class="n">IGVsession</span><span class="p">(</span><span class="n">files</span><span class="o">=</span><span class="n">myurls</span><span class="p">,</span> 
                      <span class="n">sessionFile</span><span class="o">=</span><span class="s2">"session.xml"</span><span class="p">,</span> 
                      <span class="n">genome</span><span class="o">=</span><span class="s2">"A. thaliana (TAIR10)"</span><span class="p">)</span>
<span class="n">IGVload</span><span class="p">(</span><span class="n">sock</span><span class="p">,</span> <span class="n">session</span><span class="p">)</span>
<span class="n">IGVgoto</span><span class="p">(</span><span class="n">sock</span><span class="p">,</span> <span class="s1">'Chr1:45296-47019'</span><span class="p">)</span></code></pre></figure>



<div class="tags">
    
</div>

    

</div>

<hr class="shaded"/>

<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
<p>Page last updated:</span> Mon Jul  4 15:47:21 2016<br/> Site last generated: Jul 9, 2016 <br />
<!--
<p><img src="/CSHL_RNAseq/images/company_logo.png" alt="Company logo"/></p>
-->
</div>
            </div>
</footer>


    </div>
    <!-- /.row -->
</div>
<!-- /.container -->
    </div>
    <script>
    jQuery(function ($) {
        $('.post-content').annotator()
                .annotator('setupPlugins', {tokenUrl: 'http://example.com/api/token'})
    });
    </script>
</body>

</html>