---
title: Plant SECRETome Project Site
sidebar: 
last_updated: Thu Jul 21 16:14:54 2016
---

## Project Objectives

The goal of this project is to provide a foundation for the identification of
components determining different secretory cell types in plants, their
biological functions and regulatory mechanisms in the context of development
and environmental stimuli. We will provide data sets, genetic material and gene
editing technologies that will advance the studies on cell-specific gene
regulation and cellular secretory functions. We will address a major resource
gap: the lack of secretory cell-specific promoters (_e.g._, to control genes for
defense against the pathogens, manage cell-specific processes during abiotic
stress, or enhance the flux of nutrients towards seeds to improve yield). We
will train high school, community college and four-year college undergraduate,
graduate students and postdocs in cell biology, genomic-editing, genomics,
bioinformatics and collaborative interdisciplinary research.

### subsubsection
aldkfjsdlf asdsdfsdlf asdf sdklaklf;asdfasdf 

<img title="frontpage" src="mydoc/plantsecretome/frontpage_small_img1.png" width="42"><img/>

## Background

During the evolution of multicellular eukaryotes, cells specialized in
different tasks. In plants, some cells serve as primary producers whereas
others fully depend on nutrient supply and are ‘receivers’.  Prominent examples
of producers are photosynthetic cells. Embryo cells exemplify receivers that
depend on external nutrient and energy supply. Certain situations are, however,
more complex. For example, roots are generally receivers, but can serve
secretory functions (e.g., epidermal cells exude chelators and sugars and
vascular cells transport essential nutrients to shoots). Transport processes
bridge producer and receiver cells. Specifically, phloem and xylem parenchyma
facilitate phloem and xylem translocation; root pericycle cells load the xylem
with ions. Secretory activities are critical for cuticle production of the leaf
epidermis as well as suberization and lignification of the root hypodermis to
reduce water loss in dry soil or oxygen loss in water-saturated soil. Pollen
and embryos depend on maternal supply with nutrients. Maturing pollen is fed
and coated by the anther tapetum; and small molecule transfer between cells is
key to embryo development and germination. Certain organs, i.e., hydathodes and
nectaries, specialize in secretory activities. Moreover, secretory activities
can be influenced by the environment: symbionts or pathogens tap into the
host’s nutrient resources by modulating the expression profiles of host cells
and abiotic stresses perturb the ability to produce photosynthate and limit
resource availability for growth. Under these circumstances, movement of
nutrients can be dramatically altered. Here, we define ‘secretory cells’ as
those having important functions in the secretion of solutes and
macromolecules. This project will explore the key mechanistic features of
secretory cells, taking advantage of the ability to monitor cell-type specific
mRNA translation using TRAP (Translating Ribosome Affinity Purification) that
enables profiling of mRNAs and non-coding RNAs associated with ribosomes
(translatome) and CRISPR/Cas9-aided genomic-editing.


## Expected Deliverables

__Deliverables:__ (1) Translatome atlas (the entirety of ribosome-associated
mRNAs and non-coding RNAs) of secretory cells and their activities of two plant
species; (2) Cross- species characterization of secretory cells through
non-invasive gene profiling and physiological assays; (3) Genetic resources for
functional analyses, i.e. cell-specific promoters, CRISPR/Cas9-aided gene
knock-out and reporter gene knock-in strategies; (4) Detailing of novel
processes: pathways, circuits and functions of secretory cells; (5) SECRETome
web-based resources; and (6) Public education on genomic editing technologies
through forums and multimedia resources.


## Keywords

_Oryza sativa_, _Arabidopsis thaliana_, secretory cells, cell-specific
gene expression, translatome, RNA-seq, TRAP-seq, CRISPR/Cas9, genomic 
editing, transporters, transcription factors, bioinformatics






