<!DOCTYPE html>
<head>
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>GO term enrichment analysis of DEGs  | Manual</title>
<link rel="stylesheet" href="/CSHL_RNAseq/css/syntax.css">


<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="css/bootstrap.min.css">-->
<link rel="stylesheet" href="/CSHL_RNAseq/css/modern-business.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/lavish-bootstrap.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/customstyles.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/theme-blue.css">

<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="/CSHL_RNAseq/js/jquery.navgoco.min.js"></script>


<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="/CSHL_RNAseq/js/toc.js"></script>
<script src="/CSHL_RNAseq/js/customscripts.js"></script>

<link rel="shortcut icon" href="/CSHL_RNAseq/images/favicon.ico">

<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->

<link rel="alternate" type="application/rss+xml" title="" href="http://idratherbewriting.com/CSHL_RNAseq/feed.xml">




 
    <script>
        $(document).ready(function() {
            // Initialize navgoco with default options
            $("#mysidebar").navgoco({
                caretHtml: '',
                accordion: true,
                openClass: 'active', // open
                save: false, // leave false or nav highlighting doesn't work right
                cookie: {
                    name: 'navgoco',
                    expires: false,
                    path: '/'
                },
                slide: {
                    duration: 400,
                    easing: 'swing'
                }
            });

            $("#collapseAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', false);
            });

            $("#expandAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', true);
            });

        });

    </script>
    <script>
        $(function () {
            $('[data-toggle="tooltip"]').tooltip()
        })
    </script>
    

</head>
<body>
<!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="fa fa-home fa-lg navbar-brand" href="/CSHL_RNAseq/">&nbsp;<span class="projectTitle"> 2016 CSHL Course - Plant Sciences</span></a>
        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                
                
                
                <li><a href="https://meetings.cshl.edu/courses.aspx?course=c-plan&year=16" target="_blank">CSHL</a></li>
                
                
                
                <li><a href="https://github.com/tgirke/CSHL_RNAseq/tree/gh-pages" target="_blank">Github Repo</a></li>
                
                
                
                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                
                
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">R on CRAN</a></li>
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        <li><a href="https://support.rstudio.com/hc/en-us/articles/200711853-Keyboard-Shortcuts" target="_blank">RStudio Shortcuts</a></li>
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                    </ul>
                </li>
                
                
                
			<!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "Manual feedback";
var body = "I have some feedback about the GO term enrichment analysis of DEGs page: ";
body += window.location.href;
body += "";
var uri = "mailto:thomas.girke@ucr.edu?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>

		
                <!--comment out this block if you want to hide search-->
                <li>
                    <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="/CSHL_RNAseq/js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                            SimpleJekyllSearch.init({
                                searchInput: document.getElementById('search-input'),
                                resultsContainer: document.getElementById('results-container'),
                                dataSource: '/CSHL_RNAseq/search.json',
                                searchResultTemplate: '<li><a href="{url}" title="GO term enrichment analysis of DEGs">{title}</a></li>',
                    noResultsText: 'No results found.',
                            limit: 10,
                            fuzzy: true,
                    })
                    </script>
                    <!--end search-->
                </li>
            </ul>
        </div>
        </div>
        <!-- /.container -->
</nav>
<!-- Page Content -->
<div class="container">
    <div class="col-lg-12">&nbsp;</div>
    <!-- Content Row -->
    <div class="row">
        <!-- Sidebar Column -->
        <div class="col-md-3">

          












<ul id="mysidebar" class="nav">
    <li class="sidebarTitle">RNA-Seq </li>
    
    
    
        
    
    <li>
        <a href="#">Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/">Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_requirements/">Requirements</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Slides</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_slides/">Slides</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Introduction to R</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_01/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_02/">2. R Package Repositories</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_03/">3. Installation of R and Add-on Packages</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_04/">4. Getting Around</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_05/">5. Basic Syntax</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_06/">6. Data Types</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_07/">7. Data objects</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_08/">8. Important Utilities</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_09/">9. Operators and Calculations</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_10/">10. Reading and Writing External Data</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_11/">11. Useful R Functions</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_12/">12. SQLite Databases</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_13/">13. R Markdown</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_14/">14. Session Info</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_15/">15. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">R Graphics</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_1/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_2/">2. Base Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_3/">3. Grid Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_4/">4. lattice Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_5/">5. ggplot2 Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_6/">6. Specialty Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_7/">7. Genome Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_8/">8. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">NGS Workflow Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_1/">1. Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_2/">2. Getting Started</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_3/">3. More detail</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_4/">4. Workflow demo</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_5/">5. Version information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_6/">6. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">RNA-Seq Workflow</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_01/">1. Introduction</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_02/">2. Sample definitions and environment settings</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_03/">3. Read preprocessing</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_04/">4. Alignments</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_05/">5. Read distribution across genomic features</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_06/">6. Read quantification per annotation range</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_07/">7. Analysis of differentially expressed genes with edgeR</a></li>
            
            
            
            
            
            
            <li class="active"><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_08/">8. GO term enrichment analysis of DEGs</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_09/">9. Clustering and heat maps</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_10/">10. Render report in HTML and PDF format</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_11/">11. Version Information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_12/">12. Funding</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_13/">13. References</a></li>
            
            
            
            
        </ul>
        
        
        
        <!-- if you aren't using the accordion, uncomment this block:
           <p class="external">
               <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
           </p>
           -->
    </li>
</ul>
</div>

<!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
<script>$("li.active").parents('li').toggleClass("active");</script>
    <!-- Content Column -->
    <div class="col-md-9">
        <div class="post-header">
   <h1 class="post-title-main">GO term enrichment analysis of DEGs</h1>
</div>



<div class="post-content">

   

    
    
<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>

<div id="toc"></div>

    

    

    

    

  <h2 id="obtain-gene-to-go-mappings">Obtain gene-to-GO mappings</h2>

<p>The following shows how to obtain gene-to-GO mappings from <code class="highlighter-rouge">biomaRt</code> (here for <em>A.
thaliana</em>) and how to organize them for the downstream GO term
enrichment analysis. Alternatively, the gene-to-GO mappings can be
obtained for many organisms from Bioconductorâ€™s <code class="highlighter-rouge">*.db</code> genome annotation
packages or GO annotation files provided by various genome databases.
For each annotation this relatively slow preprocessing step needs to be
performed only once. Subsequently, the preprocessed data can be loaded
with the <code class="highlighter-rouge">load</code> function as shown in the next subsection.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="s2">"biomaRt"</span><span class="p">)</span>
<span class="n">listMarts</span><span class="p">()</span> <span class="c1"># To choose BioMart database
</span><span class="n">listMarts</span><span class="p">(</span><span class="n">host</span><span class="o">=</span><span class="s2">"plants.ensembl.org"</span><span class="p">)</span>
<span class="n">m</span> <span class="o">&lt;-</span> <span class="n">useMart</span><span class="p">(</span><span class="s2">"plants_mart"</span><span class="p">,</span> <span class="n">host</span><span class="o">=</span><span class="s2">"plants.ensembl.org"</span><span class="p">)</span>
<span class="n">listDatasets</span><span class="p">(</span><span class="n">m</span><span class="p">)</span>
<span class="n">m</span> <span class="o">&lt;-</span> <span class="n">useMart</span><span class="p">(</span><span class="s2">"plants_mart"</span><span class="p">,</span> <span class="n">dataset</span><span class="o">=</span><span class="s2">"athaliana_eg_gene"</span><span class="p">,</span> <span class="n">host</span><span class="o">=</span><span class="s2">"plants.ensembl.org"</span><span class="p">)</span>
<span class="n">listAttributes</span><span class="p">(</span><span class="n">m</span><span class="p">)</span> <span class="c1"># Choose data types you want to download
</span><span class="n">go</span> <span class="o">&lt;-</span> <span class="n">getBM</span><span class="p">(</span><span class="n">attributes</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="s2">"go_accession"</span><span class="p">,</span> <span class="s2">"tair_locus"</span><span class="p">,</span> <span class="s2">"go_namespace_1003"</span><span class="p">),</span> <span class="n">mart</span><span class="o">=</span><span class="n">m</span><span class="p">)</span>
<span class="n">go</span> <span class="o">&lt;-</span> <span class="n">go</span><span class="p">[</span><span class="n">go</span><span class="p">[,</span><span class="m">3</span><span class="p">]</span><span class="o">!=</span><span class="s2">""</span><span class="p">,];</span> <span class="n">go</span><span class="p">[,</span><span class="m">3</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="n">as.character</span><span class="p">(</span><span class="n">go</span><span class="p">[,</span><span class="m">3</span><span class="p">])</span>
<span class="n">go</span><span class="p">[</span><span class="n">go</span><span class="p">[,</span><span class="m">3</span><span class="p">]</span><span class="o">==</span><span class="s2">"molecular_function"</span><span class="p">,</span> <span class="m">3</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s2">"F"</span><span class="p">;</span> <span class="n">go</span><span class="p">[</span><span class="n">go</span><span class="p">[,</span><span class="m">3</span><span class="p">]</span><span class="o">==</span><span class="s2">"biological_process"</span><span class="p">,</span> <span class="m">3</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s2">"P"</span><span class="p">;</span> <span class="n">go</span><span class="p">[</span><span class="n">go</span><span class="p">[,</span><span class="m">3</span><span class="p">]</span><span class="o">==</span><span class="s2">"cellular_component"</span><span class="p">,</span> <span class="m">3</span><span class="p">]</span> <span class="o">&lt;-</span> <span class="s2">"C"</span>
<span class="n">go</span><span class="p">[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,]</span>
<span class="n">dir.create</span><span class="p">(</span><span class="s2">"./data/GO"</span><span class="p">)</span>
<span class="n">write.table</span><span class="p">(</span><span class="n">go</span><span class="p">,</span> <span class="s2">"data/GO/GOannotationsBiomart_mod.txt"</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">col.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">)</span>
<span class="n">catdb</span> <span class="o">&lt;-</span> <span class="n">makeCATdb</span><span class="p">(</span><span class="n">myfile</span><span class="o">=</span><span class="s2">"data/GO/GOannotationsBiomart_mod.txt"</span><span class="p">,</span> <span class="n">lib</span><span class="o">=</span><span class="n">NULL</span><span class="p">,</span> <span class="n">org</span><span class="o">=</span><span class="s2">""</span><span class="p">,</span> <span class="n">colno</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">,</span><span class="m">3</span><span class="p">),</span> <span class="n">idconv</span><span class="o">=</span><span class="n">NULL</span><span class="p">)</span>
<span class="n">save</span><span class="p">(</span><span class="n">catdb</span><span class="p">,</span> <span class="n">file</span><span class="o">=</span><span class="s2">"data/GO/catdb.RData"</span><span class="p">)</span></code></pre></figure>

<h2 id="batch-go-term-enrichment-analysis">Batch GO term enrichment analysis</h2>

<p>Apply the enrichment analysis to the DEG sets obtained the above differential
expression analysis. Note, in the following example the <code class="highlighter-rouge">FDR</code> filter is set
here to an unreasonably high value, simply because of the small size of the toy
data set used in this vignette. Batch enrichment analysis of many gene sets is
performed with the function. When <code class="highlighter-rouge">method=all</code>, it returns all GO terms passing
the p-value cutoff specified under the <code class="highlighter-rouge">cutoff</code> arguments. When <code class="highlighter-rouge">method=slim</code>,
it returns only the GO terms specified under the <code class="highlighter-rouge">myslimv</code> argument. The given
example shows how a GO slim vector for a specific organism can be obtained from
BioMart.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="s2">"biomaRt"</span><span class="p">)</span>
<span class="n">load</span><span class="p">(</span><span class="s2">"data/GO/catdb.RData"</span><span class="p">)</span>
<span class="n">DEG_list</span> <span class="o">&lt;-</span> <span class="n">filterDEGs</span><span class="p">(</span><span class="n">degDF</span><span class="o">=</span><span class="n">edgeDF</span><span class="p">,</span> <span class="n">filter</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="n">Fold</span><span class="o">=</span><span class="m">2</span><span class="p">,</span> <span class="n">FDR</span><span class="o">=</span><span class="m">50</span><span class="p">),</span> <span class="n">plot</span><span class="o">=</span><span class="n">FALSE</span><span class="p">)</span>
<span class="n">up_down</span> <span class="o">&lt;-</span> <span class="n">DEG_list</span><span class="o">$</span><span class="n">UporDown</span><span class="p">;</span> <span class="n">names</span><span class="p">(</span><span class="n">up_down</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">paste</span><span class="p">(</span><span class="n">names</span><span class="p">(</span><span class="n">up_down</span><span class="p">),</span> <span class="s2">"_up_down"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">""</span><span class="p">)</span>
<span class="n">up</span> <span class="o">&lt;-</span> <span class="n">DEG_list</span><span class="o">$</span><span class="n">Up</span><span class="p">;</span> <span class="n">names</span><span class="p">(</span><span class="n">up</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">paste</span><span class="p">(</span><span class="n">names</span><span class="p">(</span><span class="n">up</span><span class="p">),</span> <span class="s2">"_up"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">""</span><span class="p">)</span>
<span class="n">down</span> <span class="o">&lt;-</span> <span class="n">DEG_list</span><span class="o">$</span><span class="n">Down</span><span class="p">;</span> <span class="n">names</span><span class="p">(</span><span class="n">down</span><span class="p">)</span> <span class="o">&lt;-</span> <span class="n">paste</span><span class="p">(</span><span class="n">names</span><span class="p">(</span><span class="n">down</span><span class="p">),</span> <span class="s2">"_down"</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">""</span><span class="p">)</span>
<span class="n">DEGlist</span> <span class="o">&lt;-</span> <span class="n">c</span><span class="p">(</span><span class="n">up_down</span><span class="p">,</span> <span class="n">up</span><span class="p">,</span> <span class="n">down</span><span class="p">)</span>
<span class="n">DEGlist</span> <span class="o">&lt;-</span> <span class="n">DEGlist</span><span class="p">[</span><span class="n">sapply</span><span class="p">(</span><span class="n">DEGlist</span><span class="p">,</span> <span class="n">length</span><span class="p">)</span> <span class="o">&gt;</span> <span class="m">0</span><span class="p">]</span>
<span class="n">BatchResult</span> <span class="o">&lt;-</span> <span class="n">GOCluster_Report</span><span class="p">(</span><span class="n">catdb</span><span class="o">=</span><span class="n">catdb</span><span class="p">,</span> <span class="n">setlist</span><span class="o">=</span><span class="n">DEGlist</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s2">"all"</span><span class="p">,</span> <span class="n">id_type</span><span class="o">=</span><span class="s2">"gene"</span><span class="p">,</span> <span class="n">CLSZ</span><span class="o">=</span><span class="m">2</span><span class="p">,</span> <span class="n">cutoff</span><span class="o">=</span><span class="m">0.9</span><span class="p">,</span> <span class="n">gocats</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="s2">"MF"</span><span class="p">,</span> <span class="s2">"BP"</span><span class="p">,</span> <span class="s2">"CC"</span><span class="p">),</span> <span class="n">recordSpecGO</span><span class="o">=</span><span class="n">NULL</span><span class="p">)</span>
<span class="n">library</span><span class="p">(</span><span class="s2">"biomaRt"</span><span class="p">)</span>
<span class="n">m</span> <span class="o">&lt;-</span> <span class="n">useMart</span><span class="p">(</span><span class="s2">"plants_mart"</span><span class="p">,</span> <span class="n">dataset</span><span class="o">=</span><span class="s2">"athaliana_eg_gene"</span><span class="p">,</span> <span class="n">host</span><span class="o">=</span><span class="s2">"plants.ensembl.org"</span><span class="p">)</span>
<span class="n">goslimvec</span> <span class="o">&lt;-</span> <span class="n">as.character</span><span class="p">(</span><span class="n">getBM</span><span class="p">(</span><span class="n">attributes</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="s2">"goslim_goa_accession"</span><span class="p">),</span> <span class="n">mart</span><span class="o">=</span><span class="n">m</span><span class="p">)[,</span><span class="m">1</span><span class="p">])</span>
<span class="n">BatchResultslim</span> <span class="o">&lt;-</span> <span class="n">GOCluster_Report</span><span class="p">(</span><span class="n">catdb</span><span class="o">=</span><span class="n">catdb</span><span class="p">,</span> <span class="n">setlist</span><span class="o">=</span><span class="n">DEGlist</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="s2">"slim"</span><span class="p">,</span> <span class="n">id_type</span><span class="o">=</span><span class="s2">"gene"</span><span class="p">,</span> <span class="n">myslimv</span><span class="o">=</span><span class="n">goslimvec</span><span class="p">,</span> <span class="n">CLSZ</span><span class="o">=</span><span class="m">10</span><span class="p">,</span> <span class="n">cutoff</span><span class="o">=</span><span class="m">0.01</span><span class="p">,</span> <span class="n">gocats</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="s2">"MF"</span><span class="p">,</span> <span class="s2">"BP"</span><span class="p">,</span> <span class="s2">"CC"</span><span class="p">),</span> <span class="n">recordSpecGO</span><span class="o">=</span><span class="n">NULL</span><span class="p">)</span></code></pre></figure>

<h2 id="plot-batch-go-term-results">Plot batch GO term results</h2>

<p>The <code class="highlighter-rouge">data.frame</code> generated by <code class="highlighter-rouge">GOCluster</code> can be plotted with the <code class="highlighter-rouge">goBarplot</code> function. Because of the
variable size of the sample sets, it may not always be desirable to show
the results from different DEG sets in the same bar plot. Plotting
single sample sets is achieved by subsetting the input data frame as
shown in the first line of the following example.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">gos</span> <span class="o">&lt;-</span> <span class="n">BatchResultslim</span><span class="p">[</span><span class="n">grep</span><span class="p">(</span><span class="s2">"M6-V6_up_down"</span><span class="p">,</span> <span class="n">BatchResultslim</span><span class="o">$</span><span class="n">CLID</span><span class="p">),</span> <span class="p">]</span>
<span class="n">gos</span> <span class="o">&lt;-</span> <span class="n">BatchResultslim</span>
<span class="n">png</span><span class="p">(</span><span class="s2">"./results/GOslimbarplotMF.png"</span><span class="p">,</span> <span class="n">height</span><span class="o">=</span><span class="m">12</span><span class="p">,</span> <span class="n">width</span><span class="o">=</span><span class="m">12</span><span class="p">,</span> <span class="n">units</span><span class="o">=</span><span class="s2">"in"</span><span class="p">,</span> <span class="n">res</span><span class="o">=</span><span class="m">72</span><span class="p">)</span>
<span class="n">goBarplot</span><span class="p">(</span><span class="n">gos</span><span class="p">,</span> <span class="n">gocat</span><span class="o">=</span><span class="s2">"MF"</span><span class="p">)</span>
<span class="n">dev.off</span><span class="p">()</span>
<span class="n">goBarplot</span><span class="p">(</span><span class="n">gos</span><span class="p">,</span> <span class="n">gocat</span><span class="o">=</span><span class="s2">"BP"</span><span class="p">)</span>
<span class="n">goBarplot</span><span class="p">(</span><span class="n">gos</span><span class="p">,</span> <span class="n">gocat</span><span class="o">=</span><span class="s2">"CC"</span><span class="p">)</span></code></pre></figure>

<p><img src="../systemPipeRNAseq_files/GOslimbarplotMF.png" alt="" /></p>
<div align="center">Figure 5: GO Slim Barplot for MF Ontology</div>



<div class="tags">
    
</div>

    

</div>

<hr class="shaded"/>

<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
<p>Page last updated:</span> Sat Jul  9 09:06:03 2016<br/> Site last generated: Jul 9, 2016 <br />
<!--
<p><img src="/CSHL_RNAseq/images/company_logo.png" alt="Company logo"/></p>
-->
</div>
            </div>
</footer>


    </div>
    <!-- /.row -->
</div>
<!-- /.container -->
    </div>
    <script>
    jQuery(function ($) {
        $('.post-content').annotator()
                .annotator('setupPlugins', {tokenUrl: 'http://example.com/api/token'})
    });
    </script>
</body>

</html>