<!DOCTYPE html>
<head>
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Alignments  | Manual</title>
<link rel="stylesheet" href="/CSHL_RNAseq/css/syntax.css">


<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="css/bootstrap.min.css">-->
<link rel="stylesheet" href="/CSHL_RNAseq/css/modern-business.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/lavish-bootstrap.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/customstyles.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/theme-blue.css">

<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="/CSHL_RNAseq/js/jquery.navgoco.min.js"></script>


<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="/CSHL_RNAseq/js/toc.js"></script>
<script src="/CSHL_RNAseq/js/customscripts.js"></script>

<link rel="shortcut icon" href="/CSHL_RNAseq/images/favicon.ico">

<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->

<link rel="alternate" type="application/rss+xml" title="" href="http://idratherbewriting.com/CSHL_RNAseq/feed.xml">




 
    <script>
        $(document).ready(function() {
            // Initialize navgoco with default options
            $("#mysidebar").navgoco({
                caretHtml: '',
                accordion: true,
                openClass: 'active', // open
                save: false, // leave false or nav highlighting doesn't work right
                cookie: {
                    name: 'navgoco',
                    expires: false,
                    path: '/'
                },
                slide: {
                    duration: 400,
                    easing: 'swing'
                }
            });

            $("#collapseAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', false);
            });

            $("#expandAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', true);
            });

        });

    </script>
    <script>
        $(function () {
            $('[data-toggle="tooltip"]').tooltip()
        })
    </script>
    

</head>
<body>
<!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="fa fa-home fa-lg navbar-brand" href="/CSHL_RNAseq/">&nbsp;<span class="projectTitle"> 2016 CSHL Course - Plant Sciences</span></a>
        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                
                
                
                <li><a href="https://meetings.cshl.edu/courses.aspx?course=c-plan&year=16" target="_blank">CSHL</a></li>
                
                
                
                <li><a href="https://github.com/tgirke/CSHL_RNAseq/tree/gh-pages" target="_blank">Github Repo</a></li>
                
                
                
                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                
                
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">R on CRAN</a></li>
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        <li><a href="https://support.rstudio.com/hc/en-us/articles/200711853-Keyboard-Shortcuts" target="_blank">RStudio Shortcuts</a></li>
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                    </ul>
                </li>
                
                
                
			<!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "Manual feedback";
var body = "I have some feedback about the Alignments page: ";
body += window.location.href;
body += "";
var uri = "mailto:thomas.girke@ucr.edu?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>

		
                <!--comment out this block if you want to hide search-->
                <li>
                    <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="/CSHL_RNAseq/js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                            SimpleJekyllSearch.init({
                                searchInput: document.getElementById('search-input'),
                                resultsContainer: document.getElementById('results-container'),
                                dataSource: '/CSHL_RNAseq/search.json',
                                searchResultTemplate: '<li><a href="{url}" title="Alignments">{title}</a></li>',
                    noResultsText: 'No results found.',
                            limit: 10,
                            fuzzy: true,
                    })
                    </script>
                    <!--end search-->
                </li>
            </ul>
        </div>
        </div>
        <!-- /.container -->
</nav>
<!-- Page Content -->
<div class="container">
    <div class="col-lg-12">&nbsp;</div>
    <!-- Content Row -->
    <div class="row">
        <!-- Sidebar Column -->
        <div class="col-md-3">

          












<ul id="mysidebar" class="nav">
    <li class="sidebarTitle">RNA-Seq </li>
    
    
    
        
    
    <li>
        <a href="#">Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/">Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_requirements/">Requirements</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Slides</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_slides/">Slides</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Introduction to R</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_01/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_02/">2. R Package Repositories</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_03/">3. Installation of R and Add-on Packages</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_04/">4. Getting Around</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_05/">5. Basic Syntax</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_06/">6. Data Types</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_07/">7. Data objects</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_08/">8. Important Utilities</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_09/">9. Operators and Calculations</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_10/">10. Reading and Writing External Data</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_11/">11. Useful R Functions</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_12/">12. SQLite Databases</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_13/">13. R Markdown</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_14/">14. Session Info</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_15/">15. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">R Graphics</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_1/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_2/">2. Base Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_3/">3. Grid Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_4/">4. lattice Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_5/">5. ggplot2 Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_6/">6. Specialty Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_7/">7. Genome Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_8/">8. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">NGS Workflow Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_1/">1. Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_2/">2. Getting Started</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_3/">3. More detail</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_4/">4. Workflow demo</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_5/">5. Version information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_6/">6. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">RNA-Seq Workflow</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_01/">1. Introduction</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_02/">2. Sample definitions and environment settings</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_03/">3. Read preprocessing</a></li>
            
            
            
            
            
            
            <li class="active"><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_04/">4. Alignments</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_05/">5. Read distribution across genomic features</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_06/">6. Read quantification per annotation range</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_07/">7. Analysis of differentially expressed genes with edgeR</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_08/">8. GO term enrichment analysis of DEGs</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_09/">9. Clustering and heat maps</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_10/">10. Render report in HTML and PDF format</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_11/">11. Version Information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_12/">12. Funding</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_13/">13. References</a></li>
            
            
            
            
        </ul>
        
        
        
        <!-- if you aren't using the accordion, uncomment this block:
           <p class="external">
               <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
           </p>
           -->
    </li>
</ul>
</div>

<!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
<script>$("li.active").parents('li').toggleClass("active");</script>
    <!-- Content Column -->
    <div class="col-md-9">
        <div class="post-header">
   <h1 class="post-title-main">Alignments</h1>
</div>



<div class="post-content">

   

    
    
<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>

<div id="toc"></div>

    

    

    

    

  <h2 id="read-mapping-with-bowtie2tophat2">Read mapping with <code class="highlighter-rouge">Bowtie2/Tophat2</code></h2>
<p>The NGS reads of this project will be aligned against the reference genome
sequence using <code class="highlighter-rouge">Bowtie2/TopHat2</code> (Kim et al., 2013; Langmead et al., 2012). The
parameter settings of the aligner are defined in the <code class="highlighter-rouge">tophat.param</code>
file.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">args</span> <span class="o">&lt;-</span> <span class="n">systemArgs</span><span class="p">(</span><span class="n">sysma</span><span class="o">=</span><span class="s2">"param/tophat.param"</span><span class="p">,</span> <span class="n">mytargets</span><span class="o">=</span><span class="s2">"targets_trim.txt"</span><span class="p">)</span>
<span class="n">sysargs</span><span class="p">(</span><span class="n">args</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span> <span class="err">#</span> <span class="n">Command</span><span class="o">-</span><span class="n">line</span> <span class="n">parameters</span> <span class="k">for</span> <span class="n">first</span> <span class="n">FASTQ</span> <span class="n">file</span></code></pre></figure>

<p>Submission of alignment jobs to compute cluster, here using 72 CPU cores (18 <code class="highlighter-rouge">qsub</code> processes each with 4 CPU cores).</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">moduleload</span><span class="p">(</span><span class="n">modules</span><span class="p">(</span><span class="n">args</span><span class="p">))</span>
<span class="n">system</span><span class="p">(</span><span class="s2">"bowtie2-build ./data/tair10.fasta ./data/tair10.fasta"</span><span class="p">)</span>
<span class="n">resources</span> <span class="o">&lt;-</span> <span class="n">list</span><span class="p">(</span><span class="n">walltime</span><span class="o">=</span><span class="s2">"20:00:00"</span><span class="p">,</span> <span class="n">nodes</span><span class="o">=</span><span class="n">paste0</span><span class="p">(</span><span class="s2">"1:ppn="</span><span class="p">,</span> <span class="n">cores</span><span class="p">(</span><span class="n">args</span><span class="p">)),</span> <span class="n">memory</span><span class="o">=</span><span class="s2">"10gb"</span><span class="p">)</span>
<span class="n">reg</span> <span class="o">&lt;-</span> <span class="n">clusterRun</span><span class="p">(</span><span class="n">args</span><span class="p">,</span> <span class="n">conffile</span><span class="o">=</span><span class="s2">".BatchJobs.R"</span><span class="p">,</span> <span class="n">template</span><span class="o">=</span><span class="s2">"torque.tmpl"</span><span class="p">,</span> <span class="n">Njobs</span><span class="o">=</span><span class="m">18</span><span class="p">,</span> <span class="n">runid</span><span class="o">=</span><span class="s2">"01"</span><span class="p">,</span> 
                  <span class="n">resourceList</span><span class="o">=</span><span class="n">resources</span><span class="p">)</span>
<span class="n">waitForJobs</span><span class="p">(</span><span class="n">reg</span><span class="p">)</span></code></pre></figure>

<h2 id="alignment-with-rsubread">Alignment with <code class="highlighter-rouge">Rsubread</code></h2>
<p>The following example shows how one can use within the <code class="highlighter-rouge">systemPipeR</code> environment the R-based
aligner <code class="highlighter-rouge">Rsubread</code> or other R-based functions that read from input files and write to output files.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">Rsubread</span><span class="p">)</span>
<span class="n">args</span> <span class="o">&lt;-</span> <span class="n">systemArgs</span><span class="p">(</span><span class="n">sysma</span><span class="o">=</span><span class="s2">"param/rsubread.param"</span><span class="p">,</span> <span class="n">mytargets</span><span class="o">=</span><span class="s2">"targets.txt"</span><span class="p">)</span>
<span class="n">buildindex</span><span class="p">(</span><span class="n">basename</span><span class="o">=</span><span class="n">reference</span><span class="p">(</span><span class="n">args</span><span class="p">),</span> <span class="n">reference</span><span class="o">=</span><span class="n">reference</span><span class="p">(</span><span class="n">args</span><span class="p">))</span> <span class="c1"># Build indexed reference genome
</span><span class="n">align</span><span class="p">(</span><span class="n">index</span><span class="o">=</span><span class="n">reference</span><span class="p">(</span><span class="n">args</span><span class="p">),</span> <span class="n">readfile1</span><span class="o">=</span><span class="n">infile1</span><span class="p">(</span><span class="n">args</span><span class="p">),</span> <span class="n">input_format</span><span class="o">=</span><span class="s2">"FASTQ"</span><span class="p">,</span>
      <span class="n">output_file</span><span class="o">=</span><span class="n">outfile1</span><span class="p">(</span><span class="n">args</span><span class="p">),</span> <span class="n">output_format</span><span class="o">=</span><span class="s2">"SAM"</span><span class="p">,</span> <span class="n">nthreads</span><span class="o">=</span><span class="m">2</span><span class="p">)</span>
<span class="k">for</span><span class="p">(</span><span class="n">i</span> <span class="k">in</span> <span class="n">seq</span><span class="p">(</span><span class="n">along</span><span class="o">=</span><span class="n">outfile1</span><span class="p">(</span><span class="n">args</span><span class="p">)))</span> <span class="n">asBam</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="n">outfile1</span><span class="p">(</span><span class="n">args</span><span class="p">)[</span><span class="n">i</span><span class="p">],</span> <span class="n">destination</span><span class="o">=</span><span class="n">gsub</span><span class="p">(</span><span class="s2">".sam"</span><span class="p">,</span> <span class="s2">""</span><span class="p">,</span> <span class="n">outfile1</span><span class="p">(</span><span class="n">args</span><span class="p">)[</span><span class="n">i</span><span class="p">]),</span> <span class="n">overwrite</span><span class="o">=</span><span class="n">TRUE</span><span class="p">,</span> <span class="n">indexDestination</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span>
<span class="n">unlink</span><span class="p">(</span><span class="n">outfile1</span><span class="p">(</span><span class="n">args</span><span class="p">));</span> <span class="n">unlink</span><span class="p">(</span><span class="n">paste0</span><span class="p">(</span><span class="n">outfile1</span><span class="p">(</span><span class="n">args</span><span class="p">),</span><span class="s2">".indel"</span><span class="p">))</span></code></pre></figure>

<h2 id="read-mapping-with-hisat2">Read mapping with <code class="highlighter-rouge">HISAT2</code></h2>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">args</span> <span class="o">&lt;-</span> <span class="n">systemArgs</span><span class="p">(</span><span class="n">sysma</span><span class="o">=</span><span class="s2">"param/hisat2.param"</span><span class="p">,</span> <span class="n">mytargets</span><span class="o">=</span><span class="s2">"targets.txt"</span><span class="p">)</span>
<span class="c1"># args &lt;- systemArgs(sysma="param/hisat2.param", mytargets="targets_trim.txt")
</span><span class="n">sysargs</span><span class="p">(</span><span class="n">args</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span> <span class="c1"># Command-line parameters for first FASTQ file
</span><span class="n">moduleload</span><span class="p">(</span><span class="n">modules</span><span class="p">(</span><span class="n">args</span><span class="p">))</span>
<span class="n">system</span><span class="p">(</span><span class="s2">"hisat2-build ./data/tair10.fasta ./data/tair10.fasta"</span><span class="p">)</span>
<span class="n">runCommandline</span><span class="p">(</span><span class="n">args</span><span class="o">=</span><span class="n">args</span><span class="p">)</span></code></pre></figure>

<p>Check whether all BAM files have been created</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">file.exists</span><span class="p">(</span><span class="n">outpaths</span><span class="p">(</span><span class="n">args</span><span class="p">))</span></code></pre></figure>

<h2 id="read-and-alignment-stats">Read and alignment stats</h2>

<p>The following provides an overview of the number of reads in each sample
and how many of them aligned to the reference.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">args</span> <span class="o">&lt;-</span> <span class="n">systemArgs</span><span class="p">(</span><span class="n">sysma</span><span class="o">=</span><span class="s2">"param/hisat2.param"</span><span class="p">,</span> <span class="n">mytargets</span><span class="o">=</span><span class="s2">"targets.txt"</span><span class="p">)</span>
<span class="n">read_statsDF</span> <span class="o">&lt;-</span> <span class="n">alignStats</span><span class="p">(</span><span class="n">args</span><span class="o">=</span><span class="n">args</span><span class="p">)</span> 
<span class="n">write.table</span><span class="p">(</span><span class="n">read_statsDF</span><span class="p">,</span> <span class="s2">"results/alignStats.xls"</span><span class="p">,</span> <span class="n">row.names</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">quote</span><span class="o">=</span><span class="n">FALSE</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s2">"\t"</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">read.table</span><span class="p">(</span><span class="n">system.file</span><span class="p">(</span><span class="s2">"extdata"</span><span class="p">,</span> <span class="s2">"alignStats.xls"</span><span class="p">,</span> <span class="n">package</span><span class="o">=</span><span class="s2">"systemPipeR"</span><span class="p">),</span> <span class="n">header</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">,]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##   FileName Nreads2x Nalign Perc_Aligned Nalign_Primary Perc_Aligned_Primary
## 1      M1A   192918 177961     92.24697         177961             92.24697
## 2      M1B   197484 159378     80.70426         159378             80.70426
## 3      A1A   189870 176055     92.72397         176055             92.72397
## 4      A1B   188854 147768     78.24457         147768             78.24457</code></pre></figure>

<h2 id="create-symbolic-links-for-viewing-bam-files-in-igv">Create symbolic links for viewing BAM files in IGV</h2>

<p>The <code class="highlighter-rouge">symLink2bam</code> function creates symbolic links to view the BAM alignment files in a
genome browser such as IGV. The corresponding URLs are written to a file
with a path specified under <code class="highlighter-rouge">urlfile</code> in the <code class="highlighter-rouge">results</code> directory.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">symLink2bam</span><span class="p">(</span><span class="n">sysargs</span><span class="o">=</span><span class="n">args</span><span class="p">,</span> <span class="n">htmldir</span><span class="o">=</span><span class="n">c</span><span class="p">(</span><span class="s2">"~/.html/"</span><span class="p">,</span> <span class="s2">"projects/tests/"</span><span class="p">),</span> 
            <span class="n">urlbase</span><span class="o">=</span><span class="s2">"http://biocluster.ucr.edu/~tgirke/"</span><span class="p">,</span> 
            <span class="n">urlfile</span><span class="o">=</span><span class="s2">"./results/IGVurl.txt"</span><span class="p">)</span></code></pre></figure>



<div class="tags">
    
</div>

    

</div>

<hr class="shaded"/>

<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
<p>Page last updated:</span> Sat Jul  9 09:06:03 2016<br/> Site last generated: Jul 9, 2016 <br />
<!--
<p><img src="/CSHL_RNAseq/images/company_logo.png" alt="Company logo"/></p>
-->
</div>
            </div>
</footer>


    </div>
    <!-- /.row -->
</div>
<!-- /.container -->
    </div>
    <script>
    jQuery(function ($) {
        $('.post-content').annotator()
                .annotator('setupPlugins', {tokenUrl: 'http://example.com/api/token'})
    });
    </script>
</body>

</html>