<!DOCTYPE html>
<head>
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Getting Started  | Manual</title>
<link rel="stylesheet" href="/CSHL_RNAseq/css/syntax.css">


<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="css/bootstrap.min.css">-->
<link rel="stylesheet" href="/CSHL_RNAseq/css/modern-business.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/lavish-bootstrap.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/customstyles.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/theme-blue.css">

<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="/CSHL_RNAseq/js/jquery.navgoco.min.js"></script>


<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="/CSHL_RNAseq/js/toc.js"></script>
<script src="/CSHL_RNAseq/js/customscripts.js"></script>

<link rel="shortcut icon" href="/CSHL_RNAseq/images/favicon.ico">

<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->

<link rel="alternate" type="application/rss+xml" title="" href="http://idratherbewriting.com/CSHL_RNAseq/feed.xml">




 
    <script>
        $(document).ready(function() {
            // Initialize navgoco with default options
            $("#mysidebar").navgoco({
                caretHtml: '',
                accordion: true,
                openClass: 'active', // open
                save: false, // leave false or nav highlighting doesn't work right
                cookie: {
                    name: 'navgoco',
                    expires: false,
                    path: '/'
                },
                slide: {
                    duration: 400,
                    easing: 'swing'
                }
            });

            $("#collapseAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', false);
            });

            $("#expandAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', true);
            });

        });

    </script>
    <script>
        $(function () {
            $('[data-toggle="tooltip"]').tooltip()
        })
    </script>
    

</head>
<body>
<!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="fa fa-home fa-lg navbar-brand" href="/CSHL_RNAseq/">&nbsp;<span class="projectTitle"> 2016 CSHL Course - Plant Sciences</span></a>
        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                
                
                
                <li><a href="https://meetings.cshl.edu/courses.aspx?course=c-plan&year=16" target="_blank">CSHL</a></li>
                
                
                
                <li><a href="https://github.com/tgirke/CSHL_RNAseq/tree/gh-pages" target="_blank">Github Repo</a></li>
                
                
                
                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                
                
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">R on CRAN</a></li>
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        <li><a href="https://support.rstudio.com/hc/en-us/articles/200711853-Keyboard-Shortcuts" target="_blank">RStudio Shortcuts</a></li>
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                    </ul>
                </li>
                
                
                
			<!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "Manual feedback";
var body = "I have some feedback about the Getting Started page: ";
body += window.location.href;
body += "";
var uri = "mailto:thomas.girke@ucr.edu?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>

		
                <!--comment out this block if you want to hide search-->
                <li>
                    <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="/CSHL_RNAseq/js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                            SimpleJekyllSearch.init({
                                searchInput: document.getElementById('search-input'),
                                resultsContainer: document.getElementById('results-container'),
                                dataSource: '/CSHL_RNAseq/search.json',
                                searchResultTemplate: '<li><a href="{url}" title="Getting Started">{title}</a></li>',
                    noResultsText: 'No results found.',
                            limit: 10,
                            fuzzy: true,
                    })
                    </script>
                    <!--end search-->
                </li>
            </ul>
        </div>
        </div>
        <!-- /.container -->
</nav>
<!-- Page Content -->
<div class="container">
    <div class="col-lg-12">&nbsp;</div>
    <!-- Content Row -->
    <div class="row">
        <!-- Sidebar Column -->
        <div class="col-md-3">

          












<ul id="mysidebar" class="nav">
    <li class="sidebarTitle">RNA-Seq </li>
    
    
    
        
    
    <li>
        <a href="#">Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/">Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_requirements/">Requirements</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Slides</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_slides/">Slides</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Introduction to R</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_01/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_02/">2. R Package Repositories</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_03/">3. Installation of R and Add-on Packages</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_04/">4. Getting Around</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_05/">5. Basic Syntax</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_06/">6. Data Types</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_07/">7. Data objects</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_08/">8. Important Utilities</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_09/">9. Operators and Calculations</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_10/">10. Reading and Writing External Data</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_11/">11. Useful R Functions</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_12/">12. SQLite Databases</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_13/">13. R Markdown</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_14/">14. Session Info</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_15/">15. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">R Graphics</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_1/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_2/">2. Base Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_3/">3. Grid Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_4/">4. lattice Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_5/">5. ggplot2 Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_6/">6. Specialty Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_7/">7. Genome Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_8/">8. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">NGS Workflow Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_1/">1. Background</a></li>
            
            
            
            
            
            
            <li class="active"><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_2/">2. Getting Started</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_3/">3. More detail</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_4/">4. Workflow demo</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_5/">5. Version information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_6/">6. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">RNA-Seq Workflow</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_01/">1. Introduction</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_02/">2. Sample definitions and environment settings</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_03/">3. Read preprocessing</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_04/">4. Alignments</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_05/">5. Read distribution across genomic features</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_06/">6. Read quantification per annotation range</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_07/">7. Analysis of differentially expressed genes with edgeR</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_08/">8. GO term enrichment analysis of DEGs</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_09/">9. Clustering and heat maps</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_10/">10. Render report in HTML and PDF format</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_11/">11. Version Information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_12/">12. Funding</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_13/">13. References</a></li>
            
            
            
            
        </ul>
        
        
        
        <!-- if you aren't using the accordion, uncomment this block:
           <p class="external">
               <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
           </p>
           -->
    </li>
</ul>
</div>

<!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
<script>$("li.active").parents('li').toggleClass("active");</script>
    <!-- Content Column -->
    <div class="col-md-9">
        <div class="post-header">
   <h1 class="post-title-main">Getting Started</h1>
</div>



<div class="post-content">

   

    
    
<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>

<div id="toc"></div>

    

    

    

    

  <h2 id="installation">Installation</h2>

<p>The R software for running <a href="http://www.bioconductor.org/packages/devel/bioc/html/systemPipeR.html"><em><code class="highlighter-rouge">systemPipeR</code></em></a> can be downloaded from <a href="http://cran.at.r-project.org/"><em>CRAN</em></a>. The <em><code class="highlighter-rouge">systemPipeR</code></em> environment can be installed from the R console using the <em><code class="highlighter-rouge">biocLite</code></em> install command. The associated data package <a href="https://github.com/tgirke/systemPipeRdata"><em><code class="highlighter-rouge">systemPipeRdata</code></em></a> can be installed the same way. The latter is a helper package for generating <em><code class="highlighter-rouge">systemPipeR</code></em> workflow environments with a single command containing all parameter files and sample data required to quickly test and run workflows.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">source</span><span class="p">(</span><span class="s2">"http://bioconductor.org/biocLite.R"</span><span class="p">)</span> <span class="c1"># Sources the biocLite.R installation script 
</span><span class="n">biocLite</span><span class="p">(</span><span class="s2">"systemPipeR"</span><span class="p">)</span> <span class="c1"># Installs systemPipeR 
</span><span class="n">biocLite</span><span class="p">(</span><span class="s2">"systemPipeRdata"</span><span class="p">)</span> <span class="err">#</span> <span class="n">Installs</span> <span class="n">systemPipeRdata</span></code></pre></figure>

<h2 id="loading-package-and-documentation">Loading package and documentation</h2>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="s2">"systemPipeR"</span><span class="p">)</span> <span class="c1"># Loads the package
</span><span class="n">library</span><span class="p">(</span><span class="n">help</span><span class="o">=</span><span class="s2">"systemPipeR"</span><span class="p">)</span> <span class="c1"># Lists package info
</span><span class="n">vignette</span><span class="p">(</span><span class="s2">"systemPipeR"</span><span class="p">)</span> <span class="err">#</span> <span class="n">Opens</span> <span class="n">vignette</span></code></pre></figure>

<h2 id="load-sample-data-and-workflow-templates">Load sample data and workflow templates</h2>
<p>The mini sample FASTQ files used by this overview vignette as well as the associated workflow reporting vignettes can be loaded via the <em><code class="highlighter-rouge">systemPipeRdata</code></em> package as shown below. The chosen data set <a href="http://www.ncbi.nlm.nih.gov/sra/?term=SRP010938"><code class="highlighter-rouge">SRP010938</code></a> contains 18 paired-end (PE) read sets from <em>Arabidposis thaliana</em> (Howard et al., 2013). To minimize processing time during testing, each FASTQ file has been subsetted to 90,000-100,000 randomly sampled PE reads that map to the first 100,000 nucleotides of each chromosome of the <em>A. thalina</em> genome. The corresponding reference genome sequence (FASTA) and its GFF annotion files (provided in the same download) have been truncated accordingly. This way the entire test sample data set requires less than 200MB disk storage space. A PE read set has been chosen for this test data set for flexibility, because it can be used for testing both types of analysis routines requiring either SE (single end) reads or PE reads.</p>

<p>The following generates a fully populated <em><code class="highlighter-rouge">systemPipeR</code></em> workflow environment (here for RNA-Seq) in the current working directory of an R session. At this time the package includes workflow templates for RNA-Seq, ChIP-Seq, VAR-Seq and Ribo-Seq. Templates for additional NGS applications will be provided in the future.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">systemPipeRdata</span><span class="p">)</span>
<span class="n">genWorkenvir</span><span class="p">(</span><span class="n">workflow</span><span class="o">=</span><span class="s2">"rnaseq"</span><span class="p">,</span> <span class="n">bam</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span>
<span class="n">setwd</span><span class="p">(</span><span class="s2">"rnaseq"</span><span class="p">)</span></code></pre></figure>

<h2 id="download-latest-version-of-this-tutorial">Download latest version of this tutorial</h2>

<p>In case there is an updated version of this tutorial, download its <code class="highlighter-rouge">systemPipeRNAseq.Rmd</code> source.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">download.file</span><span class="p">(</span><span class="s2">"https://raw.githubusercontent.com/tgirke/CSHL_RNAseq/gh-pages/_vignettes/05_Rworkflows/systemPipeR.Rmd"</span><span class="p">,</span> <span class="s2">"systemPipeR.Rmd"</span><span class="p">)</span></code></pre></figure>

<p>Now open the R markdown script <code class="highlighter-rouge">systemPipeRNAseq.Rmd</code>in your R IDE (<em>e.g.</em> vim-r or RStudio) and 
run the workflow as outlined below.</p>

<p>If you are on Windows please run the following command to move all input files into the right location. This problem will be fixed next week.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">file.copy</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s2">"data/fastq/"</span><span class="p">,</span> <span class="s2">"*"</span><span class="p">,</span> <span class="n">full.names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">),</span> <span class="s2">"data"</span><span class="p">)</span>
<span class="n">file.copy</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s2">"data/annotation/"</span><span class="p">,</span> <span class="s2">"*"</span><span class="p">,</span> <span class="n">full.names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">),</span> <span class="s2">"data"</span><span class="p">,</span> <span class="n">recursive</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span>
<span class="n">file.copy</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s2">"results/bam/"</span><span class="p">,</span> <span class="s2">"*"</span><span class="p">,</span> <span class="n">full.names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">),</span> <span class="s2">"results"</span><span class="p">)</span></code></pre></figure>

<p>The working environment of the sample data loaded in the previous step contains the following preconfigured directory structure. Directory names are indicated in  <span style="color:grey"><em><strong>grey</strong></em></span>. Users can change this structure as needed, but need to adjust the code in their workflows accordingly.</p>

<ul>
  <li><span style="color:grey"><em><strong>workflow/</strong></em></span> (<em>e.g.</em> <em>rnaseq/</em>)
    <ul>
      <li>This is the directory of the R session running the workflow.</li>
      <li>Run script ( <em>*.Rnw</em> or <em>*.Rmd</em>) and sample annotation (<em>targets.txt</em>) files are located here.</li>
      <li>Note, this directory can have any name (<em>e.g.</em> <span style="color:grey"><em><strong>rnaseq</strong></em></span>, <span style="color:grey"><em><strong>varseq</strong></em></span>). Changing its name does not require any modifications in the run script(s).</li>
      <li>Important subdirectories:
        <ul>
          <li><span style="color:grey"><em><strong>param/</strong></em></span>
            <ul>
              <li>Stores parameter files such as: <em>*.param</em>, <em>*.tmpl</em> and <em>*_run.sh</em>.</li>
            </ul>
          </li>
          <li><span style="color:grey"><em><strong>data/</strong></em> </span>
            <ul>
              <li>FASTQ samples</li>
              <li>Reference FASTA file</li>
              <li>Annotations</li>
              <li>etc.</li>
            </ul>
          </li>
          <li><span style="color:grey"><em><strong>results/</strong></em></span>
            <ul>
              <li>Alignment, variant and peak files (BAM, VCF, BED)</li>
              <li>Tabular result files</li>
              <li>Images and plots</li>
              <li>etc.</li>
            </ul>
          </li>
        </ul>
      </li>
    </ul>
  </li>
</ul>

<p>The following parameter files are included in each workflow template:</p>

<ol>
  <li><em><code class="highlighter-rouge">targets.txt</code></em>: initial one provided by user; downstream <em><code class="highlighter-rouge">targets_*.txt</code></em> files are generated automatically</li>
  <li><em><code class="highlighter-rouge">*.param</code></em>: defines parameter for input/output file operations, <em>e.g.</em> <em><code class="highlighter-rouge">trim.param</code></em>, <em><code class="highlighter-rouge">bwa.param</code></em>, <em><code class="highlighter-rouge">vartools.parm</code></em>, …</li>
  <li><em><code class="highlighter-rouge">*_run.sh</code></em>: optional bash script, <em>e.g.</em>: <em><code class="highlighter-rouge">gatk_run.sh</code></em></li>
  <li>Compute cluster environment (skip on single machine):
    <ul>
      <li><em><code class="highlighter-rouge">.BatchJobs</code></em>: defines type of scheduler for <em><code class="highlighter-rouge">BatchJobs</code></em></li>
      <li><em><code class="highlighter-rouge">*.tmpl</code></em>: specifies parameters of scheduler used by a system, <em>e.g.</em> Torque, SGE, StarCluster, Slurm, etc.</li>
    </ul>
  </li>
</ol>

<h2 id="structure-of-targets-file">Structure of <em><code class="highlighter-rouge">targets</code></em> file</h2>

<p>The <em><code class="highlighter-rouge">targets</code></em> file defines all input files (<em>e.g.</em> FASTQ, BAM, BCF) and sample comparisons of an analysis workflow. The following shows the format of a sample <em><code class="highlighter-rouge">targets</code></em> file included in the package. It also can be viewed and downloaded from <em><code class="highlighter-rouge">systemPipeR</code></em>’s GitHub repository <a href="https://github.com/tgirke/systemPipeR/blob/master/inst/extdata/targets.txt">here</a>. In a target file with a single type of input files, here FASTQ files of single end (SE) reads, the first three columns are mandatory including their column names, while it is four mandatory columns for FASTQ files of PE reads. All subsequent columns are optional and any number of additional columns can be added as needed.</p>

<h3 id="structure-of-targets-file-for-single-end-se-samples">Structure of <em><code class="highlighter-rouge">targets</code></em> file for single end (SE) samples</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">systemPipeR</span><span class="p">)</span>
<span class="n">targetspath</span> <span class="o">&lt;-</span> <span class="n">system.file</span><span class="p">(</span><span class="s2">"extdata"</span><span class="p">,</span> <span class="s2">"targets.txt"</span><span class="p">,</span> <span class="n">package</span><span class="o">=</span><span class="s2">"systemPipeR"</span><span class="p">)</span> 
<span class="n">read.delim</span><span class="p">(</span><span class="n">targetspath</span><span class="p">,</span> <span class="n">comment.char</span> <span class="o">=</span> <span class="s2">"#"</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##                    FileName SampleName Factor SampleLong Experiment        Date
## 1  ./data/SRR446027_1.fastq        M1A     M1  Mock.1h.A          1 23-Mar-2012
## 2  ./data/SRR446028_1.fastq        M1B     M1  Mock.1h.B          1 23-Mar-2012
## 3  ./data/SRR446029_1.fastq        A1A     A1   Avr.1h.A          1 23-Mar-2012
## 4  ./data/SRR446030_1.fastq        A1B     A1   Avr.1h.B          1 23-Mar-2012
## 5  ./data/SRR446031_1.fastq        V1A     V1   Vir.1h.A          1 23-Mar-2012
## 6  ./data/SRR446032_1.fastq        V1B     V1   Vir.1h.B          1 23-Mar-2012
## 7  ./data/SRR446033_1.fastq        M6A     M6  Mock.6h.A          1 23-Mar-2012
## 8  ./data/SRR446034_1.fastq        M6B     M6  Mock.6h.B          1 23-Mar-2012
## 9  ./data/SRR446035_1.fastq        A6A     A6   Avr.6h.A          1 23-Mar-2012
## 10 ./data/SRR446036_1.fastq        A6B     A6   Avr.6h.B          1 23-Mar-2012
## 11 ./data/SRR446037_1.fastq        V6A     V6   Vir.6h.A          1 23-Mar-2012
## 12 ./data/SRR446038_1.fastq        V6B     V6   Vir.6h.B          1 23-Mar-2012
## 13 ./data/SRR446039_1.fastq       M12A    M12 Mock.12h.A          1 23-Mar-2012
## 14 ./data/SRR446040_1.fastq       M12B    M12 Mock.12h.B          1 23-Mar-2012
## 15 ./data/SRR446041_1.fastq       A12A    A12  Avr.12h.A          1 23-Mar-2012
## 16 ./data/SRR446042_1.fastq       A12B    A12  Avr.12h.B          1 23-Mar-2012
## 17 ./data/SRR446043_1.fastq       V12A    V12  Vir.12h.A          1 23-Mar-2012
## 18 ./data/SRR446044_1.fastq       V12B    V12  Vir.12h.B          1 23-Mar-2012</code></pre></figure>

<p>To work with custom data, users need to generate a <em><code class="highlighter-rouge">targets</code></em> file containing the paths to their own FASTQ files and then provide under <em><code class="highlighter-rouge">targetspath</code></em> the path to the corresponding <em><code class="highlighter-rouge">targets</code></em> file.</p>

<h3 id="structure-of-targets-file-for-paired-end-pe-samples">Structure of <em><code class="highlighter-rouge">targets</code></em> file for paired end (PE) samples</h3>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">targetspath</span> <span class="o">&lt;-</span> <span class="n">system.file</span><span class="p">(</span><span class="s2">"extdata"</span><span class="p">,</span> <span class="s2">"targetsPE.txt"</span><span class="p">,</span> <span class="n">package</span><span class="o">=</span><span class="s2">"systemPipeR"</span><span class="p">)</span>
<span class="n">read.delim</span><span class="p">(</span><span class="n">targetspath</span><span class="p">,</span> <span class="n">comment.char</span> <span class="o">=</span> <span class="s2">"#"</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">2</span><span class="p">,</span><span class="m">1</span><span class="o">:</span><span class="m">6</span><span class="p">]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##                  FileName1                FileName2 SampleName Factor SampleLong Experiment
## 1 ./data/SRR446027_1.fastq ./data/SRR446027_2.fastq        M1A     M1  Mock.1h.A          1
## 2 ./data/SRR446028_1.fastq ./data/SRR446028_2.fastq        M1B     M1  Mock.1h.B          1</code></pre></figure>

<h3 id="sample-comparisons">Sample comparisons</h3>
<p>Sample comparisons are defined in the header lines of the <em><code class="highlighter-rouge">targets</code></em> file starting with ‘<code class="highlighter-rouge"># &lt;CMP&gt;</code>’.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">readLines</span><span class="p">(</span><span class="n">targetspath</span><span class="p">)[</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] "# Project ID: Arabidopsis - Pseudomonas alternative splicing study (SRA: SRP010938; PMID: 24098335)"                                                                              
## [2] "# The following line(s) allow to specify the contrasts needed for comparative analyses, such as DEG identification. All possible comparisons can be specified with 'CMPset: ALL'."
## [3] "# &lt;CMP&gt; CMPset1: M1-A1, M1-V1, A1-V1, M6-A6, M6-V6, A6-V6, M12-A12, M12-V12, A12-V12"                                                                                             
## [4] "# &lt;CMP&gt; CMPset2: ALL"</code></pre></figure>

<p>The function <em><code class="highlighter-rouge">readComp</code></em> imports the comparison information and stores it in a <em><code class="highlighter-rouge">list</code></em>. Alternatively, <em><code class="highlighter-rouge">readComp</code></em> can obtain the comparison information from the corresponding <em><code class="highlighter-rouge">SYSargs</code></em> object (see below). 
Note, these header lines are optional. They are mainly useful for controlling comparative analyses according to certain biological expectations, such as identifying differentially expressed genes in RNA-Seq experiments based on simple pair-wise comparisons.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">readComp</span><span class="p">(</span><span class="n">file</span><span class="o">=</span><span class="n">targetspath</span><span class="p">,</span> <span class="n">format</span><span class="o">=</span><span class="s2">"vector"</span><span class="p">,</span> <span class="n">delim</span><span class="o">=</span><span class="s2">"-"</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## $CMPset1
## [1] "M1-A1"   "M1-V1"   "A1-V1"   "M6-A6"   "M6-V6"   "A6-V6"   "M12-A12" "M12-V12" "A12-V12"
## 
## $CMPset2
##  [1] "M1-A1"   "M1-V1"   "M1-M6"   "M1-A6"   "M1-V6"   "M1-M12"  "M1-A12"  "M1-V12"  "A1-V1"  
## [10] "A1-M6"   "A1-A6"   "A1-V6"   "A1-M12"  "A1-A12"  "A1-V12"  "V1-M6"   "V1-A6"   "V1-V6"  
## [19] "V1-M12"  "V1-A12"  "V1-V12"  "M6-A6"   "M6-V6"   "M6-M12"  "M6-A12"  "M6-V12"  "A6-V6"  
## [28] "A6-M12"  "A6-A12"  "A6-V12"  "V6-M12"  "V6-A12"  "V6-V12"  "M12-A12" "M12-V12" "A12-V12"</code></pre></figure>

<h2 id="structure-of-param-file-and-sysargs-container">Structure of <em><code class="highlighter-rouge">param</code></em> file and <em><code class="highlighter-rouge">SYSargs</code></em> container</h2>
<p>The <em><code class="highlighter-rouge">param</code></em> file defines the parameters of a chosen command-line software. The following shows the format of a sample <em><code class="highlighter-rouge">param</code></em> file provided by this package.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">parampath</span> <span class="o">&lt;-</span> <span class="n">system.file</span><span class="p">(</span><span class="s2">"extdata"</span><span class="p">,</span> <span class="s2">"tophat.param"</span><span class="p">,</span> <span class="n">package</span><span class="o">=</span><span class="s2">"systemPipeR"</span><span class="p">)</span></code></pre></figure>

<p>The <em><code class="highlighter-rouge">systemArgs</code></em> function imports the definitions of both the <em><code class="highlighter-rouge">param</code></em> file
and the <em><code class="highlighter-rouge">targets</code></em> file, and stores all relevant information in a <em><code class="highlighter-rouge">SYSargs</code></em>
object (S4 class). To run the pipeline without command-line software, one can
assign <em><code class="highlighter-rouge">NULL</code></em> to <em><code class="highlighter-rouge">sysma</code></em> instead of a <em><code class="highlighter-rouge">param</code></em> file. In addition, one can
start <em><code class="highlighter-rouge">systemPipeR</code></em> workflows with pre-generated BAM files by providing a
targets file where the <em><code class="highlighter-rouge">FileName</code></em> column provides the paths to the BAM files.
Note, in the following example the usage of <em><code class="highlighter-rouge">suppressWarnings()</code></em> is only relevant for 
building this vignette. In typical workflows it should be removed.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">args</span> <span class="o">&lt;-</span> <span class="n">suppressWarnings</span><span class="p">(</span><span class="n">systemArgs</span><span class="p">(</span><span class="n">sysma</span><span class="o">=</span><span class="n">parampath</span><span class="p">,</span> <span class="n">mytargets</span><span class="o">=</span><span class="n">targetspath</span><span class="p">))</span>
<span class="n">args</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## An instance of 'SYSargs' for running 'tophat' on 18 samples</code></pre></figure>

<p>Several accessor methods are available that are named after the slot names of the <em><code class="highlighter-rouge">SYSargs</code></em> object.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">names</span><span class="p">(</span><span class="n">args</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##  [1] "targetsin"     "targetsout"    "targetsheader" "modules"       "software"      "cores"        
##  [7] "other"         "reference"     "results"       "infile1"       "infile2"       "outfile1"     
## [13] "sysargs"       "outpaths"</code></pre></figure>

<p>Of particular interest is the <em><code class="highlighter-rouge">sysargs()</code></em> method. It constructs the system
commands for running command-lined software as specified by a given <em><code class="highlighter-rouge">param</code></em>
file combined with the paths to the input samples (<em>e.g.</em> FASTQ files) provided
by a <em><code class="highlighter-rouge">targets</code></em> file. The example below shows the <em><code class="highlighter-rouge">sysargs()</code></em> output for
running TopHat2 on the first PE read sample. Evaluating the output of
<em><code class="highlighter-rouge">sysargs()</code></em> can be very helpful for designing and debugging <em><code class="highlighter-rouge">param</code></em> files
of new command-line software or changing the parameter settings of existing
ones.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">sysargs</span><span class="p">(</span><span class="n">args</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##                                                                                                                                                                                                                                                                                                                        M1A 
## "tophat -p 4 -g 1 --segment-length 25 -i 30 -I 3000 -o /home/tgirke/Dropbox/Teaching/CSHL_RNAseq/CSHL_RNAseq/_vignettes/04_Rworkflows/results/SRR446027_1.fastq.tophat /home/tgirke/Dropbox/Teaching/CSHL_RNAseq/CSHL_RNAseq/_vignettes/04_Rworkflows/data/tair10.fasta ./data/SRR446027_1.fastq ./data/SRR446027_2.fastq"</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">modules</span><span class="p">(</span><span class="n">args</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] "bowtie2/2.2.5" "tophat/2.0.14"</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">cores</span><span class="p">(</span><span class="n">args</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] 4</code></pre></figure>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">outpaths</span><span class="p">(</span><span class="n">args</span><span class="p">)[</span><span class="m">1</span><span class="p">]</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##                                                                                                                                 M1A 
## "/home/tgirke/Dropbox/Teaching/CSHL_RNAseq/CSHL_RNAseq/_vignettes/04_Rworkflows/results/SRR446027_1.fastq.tophat/accepted_hits.bam"</code></pre></figure>

<p>The content of the <em><code class="highlighter-rouge">param</code></em> file can also be returned as JSON object as follows (requires <em><code class="highlighter-rouge">rjson</code></em> package).</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">systemArgs</span><span class="p">(</span><span class="n">sysma</span><span class="o">=</span><span class="n">parampath</span><span class="p">,</span> <span class="n">mytargets</span><span class="o">=</span><span class="n">targetspath</span><span class="p">,</span> <span class="n">type</span><span class="o">=</span><span class="s2">"json"</span><span class="p">)</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">## [1] "{\"modules\":{\"n1\":\"\",\"v2\":\"bowtie2/2.2.5\",\"n1\":\"\",\"v2\":\"tophat/2.0.14\"},\"software\":{\"n1\":\"\",\"v1\":\"tophat\"},\"cores\":{\"n1\":\"-p\",\"v1\":\"4\"},\"other\":{\"n1\":\"\",\"v1\":\"-g 1 --segment-length 25 -i 30 -I 3000\"},\"outfile1\":{\"n1\":\"-o\",\"v2\":\"&lt;FileName1&gt;\",\"n3\":\"path\",\"v4\":\"./results/\",\"n5\":\"remove\",\"v1\":\"\",\"n2\":\"append\",\"v3\":\".tophat\",\"n4\":\"outextension\",\"v5\":\".tophat/accepted_hits.bam\"},\"reference\":{\"n1\":\"\",\"v1\":\"./data/tair10.fasta\"},\"infile1\":{\"n1\":\"\",\"v2\":\"&lt;FileName1&gt;\",\"n1\":\"path\",\"v2\":\"\"},\"infile2\":{\"n1\":\"\",\"v2\":\"&lt;FileName2&gt;\",\"n1\":\"path\",\"v2\":\"\"}}"</code></pre></figure>



<div class="tags">
    
</div>

    

</div>

<hr class="shaded"/>

<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
<p>Page last updated:</span> Sat Jul  9 09:20:33 2016<br/> Site last generated: Jul 9, 2016 <br />
<!--
<p><img src="/CSHL_RNAseq/images/company_logo.png" alt="Company logo"/></p>
-->
</div>
            </div>
</footer>


    </div>
    <!-- /.row -->
</div>
<!-- /.container -->
    </div>
    <script>
    jQuery(function ($) {
        $('.post-content').annotator()
                .annotator('setupPlugins', {tokenUrl: 'http://example.com/api/token'})
    });
    </script>
</body>

</html>