<!DOCTYPE html>
<head>
    <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1">
<meta name="description" content="">
<meta name="keywords" content=" ">
<title>Sample definitions and environment settings  | Manual</title>
<link rel="stylesheet" href="/CSHL_RNAseq/css/syntax.css">


<link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.5.0/css/font-awesome.min.css">
<!--<link rel="stylesheet" type="text/css" href="css/bootstrap.min.css">-->
<link rel="stylesheet" href="/CSHL_RNAseq/css/modern-business.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/lavish-bootstrap.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/customstyles.css">
<link rel="stylesheet" href="/CSHL_RNAseq/css/theme-blue.css">

<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-cookie/1.4.1/jquery.cookie.min.js"></script>
<script src="/CSHL_RNAseq/js/jquery.navgoco.min.js"></script>


<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.4/js/bootstrap.min.js"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/anchor-js/2.0.0/anchor.min.js"></script>
<script src="/CSHL_RNAseq/js/toc.js"></script>
<script src="/CSHL_RNAseq/js/customscripts.js"></script>

<link rel="shortcut icon" href="/CSHL_RNAseq/images/favicon.ico">

<!-- HTML5 Shim and Respond.js IE8 support of HTML5 elements and media queries -->
<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
<![endif]-->

<link rel="alternate" type="application/rss+xml" title="" href="http://idratherbewriting.com/CSHL_RNAseq/feed.xml">




 
    <script>
        $(document).ready(function() {
            // Initialize navgoco with default options
            $("#mysidebar").navgoco({
                caretHtml: '',
                accordion: true,
                openClass: 'active', // open
                save: false, // leave false or nav highlighting doesn't work right
                cookie: {
                    name: 'navgoco',
                    expires: false,
                    path: '/'
                },
                slide: {
                    duration: 400,
                    easing: 'swing'
                }
            });

            $("#collapseAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', false);
            });

            $("#expandAll").click(function(e) {
                e.preventDefault();
                $("#mysidebar").navgoco('toggle', true);
            });

        });

    </script>
    <script>
        $(function () {
            $('[data-toggle="tooltip"]').tooltip()
        })
    </script>
    

</head>
<body>
<!-- Navigation -->
<nav class="navbar navbar-inverse navbar-fixed-top">
    <div class="container topnavlinks">
        <div class="navbar-header">
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#bs-example-navbar-collapse-1">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            <a class="fa fa-home fa-lg navbar-brand" href="/CSHL_RNAseq/">&nbsp;<span class="projectTitle"> 2016 CSHL Course - Plant Sciences</span></a>
        </div>
        <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
            <ul class="nav navbar-nav navbar-right">
                <!-- entries without drop-downs appear here -->
                
                
                
                <li><a href="https://meetings.cshl.edu/courses.aspx?course=c-plan&year=16" target="_blank">CSHL</a></li>
                
                
                
                <li><a href="https://github.com/tgirke/CSHL_RNAseq/tree/gh-pages" target="_blank">Github Repo</a></li>
                
                
                
                <!-- entries with drop-downs appear here -->
                <!-- conditional logic to control which topnav appears for the audience defined in the configuration file.-->
                
                
                <li class="dropdown">
                    <a href="#" class="dropdown-toggle" data-toggle="dropdown">Resources<b class="caret"></b></a>
                    <ul class="dropdown-menu">
                        
                        
                        <li><a href="https://cran.r-project.org/" target="_blank">R on CRAN</a></li>
                        
                        
                        
                        <li><a href="http://bioconductor.org/" target="_blank">Bioconductor</a></li>
                        
                        
                        
                        <li><a href="https://support.rstudio.com/hc/en-us/articles/200711853-Keyboard-Shortcuts" target="_blank">RStudio Shortcuts</a></li>
                        
                        
                        
                        <li><a href="http://girke.bioinformatics.ucr.edu/" target="_blank">Girke Lab</a></li>
                        
                        
                    </ul>
                </li>
                
                
                
			<!-- Send feedback function -->
<script>
function SendLinkByMail(href) {
var subject= "Manual feedback";
var body = "I have some feedback about the Sample definitions and environment settings page: ";
body += window.location.href;
body += "";
var uri = "mailto:thomas.girke@ucr.edu?subject=";
uri += encodeURIComponent(subject);
uri += "&body=";
uri += encodeURIComponent(body);
window.location.href = uri;
}
</script>

<li><a href="javascript:(function()%7BSendLinkByMail()%3B%7D)()%3B" target="_blank"><i class="fa fa-envelope-o"></i> Feedback</a></li>

		
                <!--comment out this block if you want to hide search-->
                <li>
                    <!--start search-->
                    <div id="search-demo-container">
                        <input type="text" id="search-input" placeholder="search...">
                        <ul id="results-container"></ul>
                    </div>
                    <script src="/CSHL_RNAseq/js/jekyll-search.js" type="text/javascript"></script>
                    <script type="text/javascript">
                            SimpleJekyllSearch.init({
                                searchInput: document.getElementById('search-input'),
                                resultsContainer: document.getElementById('results-container'),
                                dataSource: '/CSHL_RNAseq/search.json',
                                searchResultTemplate: '<li><a href="{url}" title="Sample definitions and environment settings">{title}</a></li>',
                    noResultsText: 'No results found.',
                            limit: 10,
                            fuzzy: true,
                    })
                    </script>
                    <!--end search-->
                </li>
            </ul>
        </div>
        </div>
        <!-- /.container -->
</nav>
<!-- Page Content -->
<div class="container">
    <div class="col-lg-12">&nbsp;</div>
    <!-- Content Row -->
    <div class="row">
        <!-- Sidebar Column -->
        <div class="col-md-3">

          












<ul id="mysidebar" class="nav">
    <li class="sidebarTitle">RNA-Seq </li>
    
    
    
        
    
    <li>
        <a href="#">Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/">Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_requirements/">Requirements</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Slides</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_slides/">Slides</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">Introduction to R</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_01/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_02/">2. R Package Repositories</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_03/">3. Installation of R and Add-on Packages</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_04/">4. Getting Around</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_05/">5. Basic Syntax</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_06/">6. Data Types</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_07/">7. Data objects</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_08/">8. Important Utilities</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_09/">9. Operators and Calculations</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_10/">10. Reading and Writing External Data</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_11/">11. Useful R Functions</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_12/">12. SQLite Databases</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_13/">13. R Markdown</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_14/">14. Session Info</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rbasics_15/">15. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">R Graphics</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_1/">1. Overview</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_2/">2. Base Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_3/">3. Grid Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_4/">4. lattice Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_5/">5. ggplot2 Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_6/">6. Specialty Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_7/">7. Genome Graphics</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_Rgraphics_8/">8. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">NGS Workflow Overview</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_1/">1. Background</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_2/">2. Getting Started</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_3/">3. More detail</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_4/">4. Workflow demo</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_5/">5. Version information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeR_6/">6. References</a></li>
            
            
            
            
        </ul>
        
        
    
    <li>
        <a href="#">RNA-Seq Workflow</a>
        <ul>
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_01/">1. Introduction</a></li>
            
            
            
            
            
            
            <li class="active"><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_02/">2. Sample definitions and environment settings</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_03/">3. Read preprocessing</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_04/">4. Alignments</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_05/">5. Read distribution across genomic features</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_06/">6. Read quantification per annotation range</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_07/">7. Analysis of differentially expressed genes with edgeR</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_08/">8. GO term enrichment analysis of DEGs</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_09/">9. Clustering and heat maps</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_10/">10. Render report in HTML and PDF format</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_11/">11. Version Information</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_12/">12. Funding</a></li>
            
            
            
            
            
            
            <li><a href="/CSHL_RNAseq/mydoc/mydoc_systemPipeRNAseq_13/">13. References</a></li>
            
            
            
            
        </ul>
        
        
        
        <!-- if you aren't using the accordion, uncomment this block:
           <p class="external">
               <a href="#" id="collapseAll">Collapse All</a> | <a href="#" id="expandAll">Expand All</a>
           </p>
           -->
    </li>
</ul>
</div>

<!-- this highlights the active parent class in the navgoco sidebar. this is critical so that the parent expands when you're viewing a page. This must appear below the sidebar code above. Otherwise, if placed inside customscripts.js, the script runs before the sidebar code runs and the class never gets inserted.-->
<script>$("li.active").parents('li').toggleClass("active");</script>
    <!-- Content Column -->
    <div class="col-md-9">
        <div class="post-header">
   <h1 class="post-title-main">Sample definitions and environment settings</h1>
</div>



<div class="post-content">

   

    
    
<!-- this handles the automatic toc. use ## for subheads to auto-generate the on-page minitoc. if you use html tags, you must supply an ID for the heading element in order for it to appear in the minitoc. -->
<script>
$( document ).ready(function() {
  // Handler for .ready() called.

$('#toc').toc({ minimumHeaders: 0, listType: 'ul', showSpeed: 0, headers: 'h2,h3,h4' });

/* this offset helps account for the space taken up by the floating toolbar. */
$('#toc').on('click', 'a', function() {
  var target = $(this.getAttribute('href'))
    , scroll_target = target.offset().top

  $(window).scrollTop(scroll_target - 10);
  return false
})
  
});
</script>

<div id="toc"></div>

    

    

    

    

  <h2 id="environment-settings-and-input-data">Environment settings and input data</h2>

<p>Typically, the user wants to record here the sources and versions of the
reference genome sequence along with the corresponding annotations. In
the provided sample data set all data inputs are stored in a <code class="highlighter-rouge">data</code>
subdirectory and all results will be written to a separate <code class="highlighter-rouge">results</code> directory,
while the <code class="highlighter-rouge">systemPipeRNAseq.Rmd</code> script and the <code class="highlighter-rouge">targets</code> file are expected to be located in the parent
directory. The R session is expected to run from this parent directory.</p>

<p>To run this sample report, mini sample FASTQ and reference genome files
can be downloaded from
<a href="http://biocluster.ucr.edu/~tgirke/projects/systemPipeR_test_data.zip">here</a>.
The chosen data set <a href="http://www.ncbi.nlm.nih.gov/sra/?term=SRP010938">SRP010938</a>
contains 18 paired-end (PE) read sets from <em>Arabidposis thaliana</em>
(Howard et al., 2013). To minimize processing time during testing, each FASTQ
file has been subsetted to 90,000-100,000 randomly sampled PE reads that
map to the first 100,000 nucleotides of each chromosome of the <em>A.
thalina</em> genome. The corresponding reference genome sequence (FASTA) and
its GFF annotion files (provided in the same download) have been
truncated accordingly. This way the entire test sample data set is less
than 200MB in storage space. A PE read set has been chosen for this test
data set for flexibility, because it can be used for testing both types
of analysis routines requiring either SE (single end) reads or PE reads.</p>

<p>The following loads one of the available NGS workflow templates (here RNA-Seq)
into the user’s current working directory. At the moment, the package includes
workflow templates for RNA-Seq, ChIP-Seq, VAR-Seq and Ribo-Seq. Templates for
additional NGS applications will be provided in the future.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">systemPipeRdata</span><span class="p">)</span>
<span class="n">genWorkenvir</span><span class="p">(</span><span class="n">workflow</span><span class="o">=</span><span class="s2">"rnaseq"</span><span class="p">,</span> <span class="n">bam</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span>
<span class="n">setwd</span><span class="p">(</span><span class="s2">"rnaseq"</span><span class="p">)</span></code></pre></figure>

<h2 id="download-latest-version-of-this-tutorial">Download latest version of this tutorial</h2>

<p>In case there is an updated version of this tutorial, download its <code class="highlighter-rouge">systemPipeRNAseq.Rmd</code> source.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">download.file</span><span class="p">(</span><span class="s2">"https://raw.githubusercontent.com/tgirke/CSHL_RNAseq/gh-pages/_vignettes/06_RNAseq/systemPipeRNAseq.Rmd"</span><span class="p">,</span> <span class="s2">"systemPipeRNAseq.Rmd"</span><span class="p">)</span></code></pre></figure>

<p>Now open the R markdown script <code class="highlighter-rouge">systemPipeRNAseq.Rmd</code>in your R IDE (<em>e.g.</em> vim-r or RStudio) and 
run the workflow as outlined below.</p>

<p>If you are on Windows please run the following command to move all input files into the right location. This problem will be fixed next week.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">file.copy</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s2">"data/fastq/"</span><span class="p">,</span> <span class="s2">"*"</span><span class="p">,</span> <span class="n">full.names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">),</span> <span class="s2">"data"</span><span class="p">)</span>
<span class="n">file.copy</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s2">"data/annotation/"</span><span class="p">,</span> <span class="s2">"*"</span><span class="p">,</span> <span class="n">full.names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">),</span> <span class="s2">"data"</span><span class="p">,</span> <span class="n">recursive</span><span class="o">=</span><span class="n">TRUE</span><span class="p">)</span>
<span class="n">file.copy</span><span class="p">(</span><span class="n">list.files</span><span class="p">(</span><span class="s2">"results/bam/"</span><span class="p">,</span> <span class="s2">"*"</span><span class="p">,</span> <span class="n">full.names</span><span class="o">=</span><span class="n">TRUE</span><span class="p">),</span> <span class="s2">"results"</span><span class="p">)</span></code></pre></figure>

<h2 id="required-packages-and-resources">Required packages and resources</h2>

<p>The <code class="highlighter-rouge">systemPipeR</code> package needs to be loaded to perform the analysis steps shown in
this report (Girke , 2014).</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">library</span><span class="p">(</span><span class="n">systemPipeR</span><span class="p">)</span></code></pre></figure>

<p>In the workflow environments generated by <code class="highlighter-rouge">genWorkenvir</code> all data
inputs are stored in a <code class="highlighter-rouge">data/</code> directory and all analysis results will
be written to a separate <code class="highlighter-rouge">results/</code> directory, while the
<code class="highlighter-rouge">systemPipeRNAseq.Rmd</code> script and the <code class="highlighter-rouge">targets</code> file are
expected to be located in the parent directory. The R session is expected to
run from this parent directory. Additional parameter files are stored under
<code class="highlighter-rouge">param/</code>.</p>

<p>To work with real data, users want to organize their own data similarly and
substitute all test data for their own data. To rerun an established workflow
on new data, the initial <code class="highlighter-rouge">targets</code> file along with the corresponding
FASTQ files are usually the only inputs the user needs to provide.</p>

<p>If applicable users can load custom functions not provided by
<code class="highlighter-rouge">systemPipeR</code>. Skip this step if this is not the case.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">source</span><span class="p">(</span><span class="s2">"systemPipeRNAseq_Fct.R"</span><span class="p">)</span></code></pre></figure>

<h2 id="experiment-definition-provided-by-targets-file">Experiment definition provided by <code class="highlighter-rouge">targets</code> file</h2>

<p>The <code class="highlighter-rouge">targets</code> file defines all FASTQ files and sample
comparisons of the analysis workflow.</p>

<figure class="highlight"><pre><code class="language-r" data-lang="r"><span class="n">targetspath</span> <span class="o">&lt;-</span> <span class="n">system.file</span><span class="p">(</span><span class="s2">"extdata"</span><span class="p">,</span> <span class="s2">"targets.txt"</span><span class="p">,</span> <span class="n">package</span><span class="o">=</span><span class="s2">"systemPipeR"</span><span class="p">)</span>
<span class="n">targets</span> <span class="o">&lt;-</span> <span class="n">read.delim</span><span class="p">(</span><span class="n">targetspath</span><span class="p">,</span> <span class="n">comment.char</span> <span class="o">=</span> <span class="s2">"#"</span><span class="p">)[,</span><span class="m">1</span><span class="o">:</span><span class="m">4</span><span class="p">]</span>
<span class="n">targets</span></code></pre></figure>

<figure class="highlight"><pre><code class="language-txt" data-lang="txt">##                    FileName SampleName Factor SampleLong
## 1  ./data/SRR446027_1.fastq        M1A     M1  Mock.1h.A
## 2  ./data/SRR446028_1.fastq        M1B     M1  Mock.1h.B
## 3  ./data/SRR446029_1.fastq        A1A     A1   Avr.1h.A
## 4  ./data/SRR446030_1.fastq        A1B     A1   Avr.1h.B
## 5  ./data/SRR446031_1.fastq        V1A     V1   Vir.1h.A
## 6  ./data/SRR446032_1.fastq        V1B     V1   Vir.1h.B
## 7  ./data/SRR446033_1.fastq        M6A     M6  Mock.6h.A
## 8  ./data/SRR446034_1.fastq        M6B     M6  Mock.6h.B
## 9  ./data/SRR446035_1.fastq        A6A     A6   Avr.6h.A
## 10 ./data/SRR446036_1.fastq        A6B     A6   Avr.6h.B
## 11 ./data/SRR446037_1.fastq        V6A     V6   Vir.6h.A
## 12 ./data/SRR446038_1.fastq        V6B     V6   Vir.6h.B
## 13 ./data/SRR446039_1.fastq       M12A    M12 Mock.12h.A
## 14 ./data/SRR446040_1.fastq       M12B    M12 Mock.12h.B
## 15 ./data/SRR446041_1.fastq       A12A    A12  Avr.12h.A
## 16 ./data/SRR446042_1.fastq       A12B    A12  Avr.12h.B
## 17 ./data/SRR446043_1.fastq       V12A    V12  Vir.12h.A
## 18 ./data/SRR446044_1.fastq       V12B    V12  Vir.12h.B</code></pre></figure>



<div class="tags">
    
</div>

    

</div>

<hr class="shaded"/>

<footer>
            <div class="row">
                <div class="col-lg-12 footer">
               &copy;2016 UC Riverside. All rights reserved. <br />
<p>Page last updated:</span> Sat Jul  9 09:06:03 2016<br/> Site last generated: Jul 9, 2016 <br />
<!--
<p><img src="/CSHL_RNAseq/images/company_logo.png" alt="Company logo"/></p>
-->
</div>
            </div>
</footer>


    </div>
    <!-- /.row -->
</div>
<!-- /.container -->
    </div>
    <script>
    jQuery(function ($) {
        $('.post-content').annotator()
                .annotator('setupPlugins', {tokenUrl: 'http://example.com/api/token'})
    });
    </script>
</body>

</html>